{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Examples for using prunito"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.94.2'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import prunito\n",
    "prunito.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Searching and parsing UniProtKB data\n",
    "\n",
    "Prunito allows searching uniprot.org and parsing entries in flat file (text) format. First, import the relevant package:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from prunito import uniprot as up"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A simple search for all reviewed UniProtKB entries with 'tax' in their names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = up.search_reviewed('name:tax')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check how many hits this search retrieved."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result.size()\n",
    "# or\n",
    "#len(result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What would happen if a given query had many hits? By default, the maximum number of hits retrieved is 2000. This can be changed using the parameter 'limit'.\n",
    "\n",
    "Let's re-run the previous search but this time not just for reviewed (i.e. Swiss-Prot) entries but all of UniProtKB:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Partial dataset retrieved. Size: 1890. Retrieved: 1000.\n",
      "Consider increasing the limit and/or using offset.\n"
     ]
    }
   ],
   "source": [
    "huge = up.search('name:tax', limit=1000)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If the set limit is lower than the actual number of search hits, the above hint is printed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1890"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Approach 1: increase the limit of allowed hits\n",
    "new_limit = up.search('name:tax', limit=2000)\n",
    "new_limit.size()\n",
    "#"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Back to the initial result, the one with size of 41. What does 41 mean? UniProtKB entries. And these entries we would like to parse. As prunito provides functionality for both searching and parsing UniProt, one can directly iterate over the entries in a search result for convenience:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "entries = list(result)\n",
    "# or\n",
    "# for entry in result: ..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Iterate over the 41 entries, printing out primary accessions and recommended full names. Both fields are provided for convenience (unlike Biopython's Record object which does not have this)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q06507 Cyclic AMP-dependent transcription factor ATF-4\n",
      "P18848 Cyclic AMP-dependent transcription factor ATF-4\n",
      "Q9Y6D9 Mitotic spindle assembly checkpoint protein MAD1\n",
      "P47911 60S ribosomal protein L6\n",
      "P10070 Zinc finger protein GLI2\n",
      "Q0VGT2 Zinc finger protein GLI2\n",
      "O14908 PDZ domain-containing protein GIPC1\n",
      "Q02878 60S ribosomal protein L6\n",
      "O14910 Protein lin-7 homolog A\n",
      "Q10586 D site-binding protein\n",
      "P22063 Contactin-2\n",
      "Q02246 Contactin-2\n",
      "Q61330 Contactin-2\n",
      "P43489 Tumor necrosis factor receptor superfamily member 4\n",
      "Q08AM6 Protein VAC14 homolog\n",
      "P47030 Protein TAX4\n",
      "Q5R4U3 Tax1-binding protein 1 homolog\n",
      "Q4U0X7 Protein Tax-3\n",
      "P0C222 Protein Tax-1\n",
      "P23510 Tumor necrosis factor ligand superfamily member 4\n",
      "A6ZPP1 Protein TAX4\n",
      "Q2KJE0 Tax1-binding protein 1 homolog\n",
      "Q86VP1 Tax1-binding protein 1\n",
      "P14079 Protein Tax-1\n",
      "O35426 X-box-binding protein 1\n",
      "P17861 X-box-binding protein 1\n",
      "Q9DBG9 Tax1-binding protein 3\n",
      "Q6P132 Tax1-binding protein 1 homolog B\n",
      "Q4QQV1 Tax1-binding protein 3\n",
      "Q1LWB0 Tax1-binding protein 1 homolog A\n",
      "O14907 Tax1-binding protein 3\n",
      "Q6P3P1 Tax1-binding protein 1 homolog\n",
      "Q0R5R1 Protein Tax-3\n",
      "Q66HA4 Tax1-binding protein 1 homolog\n",
      "P03409 Protein Tax-1\n",
      "Q9UQ35 Serine/arginine repetitive matrix protein 2\n",
      "Q13884 Beta-1-syntrophin\n",
      "Q3UKC1 Tax1-binding protein 1 homolog\n",
      "P0C213 Protein Tax-1\n",
      "P03410 Protein Tax-2\n",
      "Q9NPB6 Partitioning defective 6 homolog alpha\n"
     ]
    }
   ],
   "source": [
    "for entry in entries:\n",
    "    print(entry.primary_accession, entry.recommended_full_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Which methods and fields are available on a Record object?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "accessions\n",
      "all_pubmed_ids\n",
      "annotation_update\n",
      "as_fasta\n",
      "comments\n",
      "created\n",
      "cross_references\n",
      "data_class\n",
      "description\n",
      "ec_numbers\n",
      "entry_name\n",
      "features\n",
      "gene_name\n",
      "host_organism\n",
      "host_taxonomy_id\n",
      "isoforms\n",
      "keywords\n",
      "molecule_type\n",
      "organelle\n",
      "organism\n",
      "organism_classification\n",
      "primary_accession\n",
      "primary_gene_name\n",
      "recommended_full_name\n",
      "references\n",
      "seqinfo\n",
      "sequence\n",
      "sequence_length\n",
      "sequence_update\n",
      "taxonomy_id\n"
     ]
    }
   ],
   "source": [
    "for item in dir(entries[0]):\n",
    "    if not item.startswith('_'):\n",
    "        print(item)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get isoforms for those entries that have them. We use the presence of a keyword, 'Alternative splicing', as a filter here."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp|Q9Y6D9-2|MD1L1_HUMAN Isoform 2 of Mitotic spindle assembly checkpoint protein MAD1 OS=Homo sapiens (Human). OX=['9606']\n",
      "MLPARGCVRKRTVWPRLARVLIVTLLTLELSYAPLPCQLSGVPYNTGDPVGRWARPCIWP\n",
      "CPWHTTINALKGRISELQWSVMDQEMRVKRLESEKQELQEQLDLQHKKCQEANQKIQELQ\n",
      "ASQEARADHEQQIKDLEQKLSLQEQDAAIVKNMKSELVRLPRLERELKQLREESAHLREM\n",
      "RETNGLLQEELEGLQRKLGRQEKMQETLVGLELENERLLAKLQSWERLDQTMGLSIRTPE\n",
      "DLSRFVVELQQRELALKDKNSAVTSSARGLEKARQQLQEELRQVSGQLLEERKKRETHEA\n",
      "LARRLQKRVLLLTKERDGMRAILGSYDSELTPAEYSPQLTRRMREAEDMVQKVHSHSAEM\n",
      "EAQLSQALEELGGQKQRADMLEMELKMLKSQSSSAEQSFLFSREEADTLRLKVEELEGER\n",
      "SRLEEEKRMLEAQLERRALQGDYDQSRTKVLHMSLNPTSVARQRLREDHSQLQAECERLR\n",
      "GLLRAMERGGTVPADLEAAAASLPSSKEVAELKKQVESAELKNQRLKEVFQTKIQEFRKA\n",
      "CYTLTGYQIDITTENQYRLTSLYAEHPGDCLIFKATSPSGSKMQLLETEFSHTVGELIEV\n",
      "HLRRQDSIPAFLSSLTLELFSRQTVA\n",
      ">sp|P10070-1|GLI2_HUMAN Isoform 1 of Zinc finger protein GLI2 OS=Homo sapiens (Human). OX=['9606']\n",
      "MALTSINATPTQLSSSSNCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPAS\n",
      "PLALTQGQVSGHGSCGCALPLSQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEY\n",
      "DTQEQLVHHINNEHIHGEKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFE\n",
      "GCSKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKI\n",
      "PGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGGPEST\n",
      "EASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGP\n",
      "GSLGDLTALDDTPPGADTSALAAPSAGGLQLRKHMTTMHRFEQLKKEKLKSLKDSCSWAG\n",
      "PTPHTRNTKLPPLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDS\n",
      "STSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSE\n",
      "ASQCSGGSGLLNLTPAQQYSLRAKYAAATGGPPPTPLPGLERMSLRTRLALLDAPERTLP\n",
      "AGCPRPLGPRRGSDGPTYGHGHAGAAPAFPHEAPGGGARRASDPVRRPDALSLPRVQRFH\n",
      "STHNVNPGPLPPCADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAG\n",
      "PEADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQR\n",
      "RMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEVSSGTVDALASQVKPPPFPQ\n",
      "GNLAVVQQKPAFGQYPGYSPQGLQASPGGLDSTQPHLQPRSGAPSQGIPRVNYMQQLRQP\n",
      "VAGSQCPGMTTTMSPHACYGQVHPQLSPSTISGALNQFPQSCSNMPAKPGHLGHPQQTEV\n",
      "APDPTTMGNRHRELGVPDSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQ\n",
      "PGFMEPQTGPMGVATAGFGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARATGHAMAAMPS\n",
      "SQETAEAVPKGAMGNMGSVPPQPPPQDAGGAPDHSMLYYYGQIHMYEQDGGLENLGSCQV\n",
      "MRSQPPQPQACQDSIQPQPLPSPGVNQVSSTVDSQLLEAPQIDFDAIMDDGDHSSLFSGA\n",
      "LSPSLLHSLSQNSSRLTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT\n",
      ">sp|P10070-2|GLI2_HUMAN Isoform 2 of Zinc finger protein GLI2 OS=Homo sapiens (Human). OX=['9606']\n",
      "MALTSINATPTQLSSSSNCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPAS\n",
      "PLALTQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHG\n",
      "EKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRLENLKTHLR\n",
      "SHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKIPGCTKRYTDPSSLRKHV\n",
      "KTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGGPESTEASSTSQAVEDCLHVRA\n",
      "IKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGAD\n",
      "TSALAAPSAGGLQLRKHMTTMHRFEQLKKEKLKSLKDSCSWAGPTPHTRNTKLPPLPGSG\n",
      "SILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSSTSTVSSAYTVSRRSSG\n",
      "ISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSEASQCSGGSGLLNLTPAQ\n",
      "QYSLRAKYAAATGGPPPTPLPGLERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPT\n",
      "YGHGHAGAAPAFPHEAPGGGARRASDPVRRPDALSLPRVQRFHSTHNVNPGPLPPCADRR\n",
      "GLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPEADLGLPEDDLVLPDD\n",
      "VVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQRRMVAADSNVGPSAPMLG\n",
      "GCQLGFGAPSSLNKNNMPVQWNEVSSGTVDALASQVKPPPFPQGNLAVVQQKPAFGQYPG\n",
      "YSPQGLQASPGGLDSTQPHLQPRSGAPSQGIPRVNYMQQLRQPVAGSQCPGMTTTMSPHA\n",
      "CYGQVHPQLSPSTISGALNQFPQSCSNMPAKPGHLGHPQQTEVAPDPTTMGNRHRELGVP\n",
      "DSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQPGFMEPQTGPMGVATAG\n",
      "FGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARATGHAMAAMPSSQETAEAVPKGAMGNMG\n",
      "SVPPQPPPQDAGGAPDHSMLYYYGQIHMYEQDGGLENLGSCQVMRSQPPQPQACQDSIQP\n",
      "QPLPSPGVNQVSSTVDSQLLEAPQIDFDAIMDDGDHSSLFSGALSPSLLHSLSQNSSRLT\n",
      "TPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT\n",
      ">sp|P10070-3|GLI2_HUMAN Isoform 3 of Zinc finger protein GLI2 OS=Homo sapiens (Human). OX=['9606']\n",
      "MALTSINATPTQLSSSSNCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPAS\n",
      "PLALTQGQVSGHGSCGCALPLSQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEY\n",
      "DTQEQLVHHINNEHIHGEKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFE\n",
      "GCSKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKI\n",
      "PGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGGPEST\n",
      "EASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGP\n",
      "GSLGDLTALDDTPPGADTSALAAPSAGGLQLRKHMTTMHRFEQLKKEKLKSLKDSCSWAG\n",
      "PTPHTRNTKLPPLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDS\n",
      "STSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSE\n",
      "ASQCSGGSGLLNLTPAQQYSLRAKYAAATGGPPPTPLPGLERMSLRTRLALLDAPERTLP\n",
      "AGCPRPLGPRRGSDGPTYGHGHAGAAPAFPHEAPGGGARRASDPVRRPDALSLPRVQRFH\n",
      "STHNVNPGPLPPCADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAG\n",
      "PEADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQR\n",
      "RMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEASATWLSGT\n",
      ">sp|P10070-4|GLI2_HUMAN Isoform 4 of Zinc finger protein GLI2 OS=Homo sapiens (Human). OX=['9606']\n",
      "MALTSINATPTQLSSSSNCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPAS\n",
      "PLALTQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHG\n",
      "EKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRLENLKTHLR\n",
      "SHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKIPGCTKRYTDPSSLRKHV\n",
      "KTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGGPESTEASSTSQAVEDCLHVRA\n",
      "IKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGAD\n",
      "TSALAAPSAGGLQLRKHMTTMHRFEQLKKEKLKSLKDSCSWAGPTPHTRNTKLPPLPGSG\n",
      "SILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSSTSTVSSAYTVSRRSSG\n",
      "ISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSEASQCSGGSGLLNLTPAQ\n",
      "QYSLRAKYAAATGGPPPTPLPGLERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPT\n",
      "YGHGHAGAAPAFPHEAPGGGARRASDPVRRPDALSLPRVQRFHSTHNVNPGPLPPCADRR\n",
      "GLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPEADLGLPEDDLVLPDD\n",
      "VVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQRRMVAADSNVGPSAPMLG\n",
      "GCQLGFGAPSSLNKNNMPVQWNEASATWLSGT\n",
      ">sp|O14908-2|GIPC1_HUMAN Isoform 2 of PDZ domain-containing protein GIPC1 OS=Homo sapiens (Human). OX=['9606']\n",
      "MFCTLNTHKVDMDKLLGGQIGLEDFIFAHVKGQRKEVEVFKSEDALGLTITDNGAGYAFI\n",
      "KRIKEGSVIDHIHLISVGDMIEAINGQSLLGCRHYEVARLLKELPRGRTFTLKLTEPRKA\n",
      "FDMISQRSAGGRPGSGPQLGTGRGTLRLRSRGPATVEDLPSAFEEKAIEKVDDLLESYMG\n",
      "IRDTELAATMVELGKDKRNPDELAEALDERLGDFAFPDEFVFDVWGAIGDAKVGRY\n",
      ">sp|Q08AM6-2|VAC14_HUMAN Isoform 2 of Protein VAC14 homolog OS=Homo sapiens (Human). OX=['9606']\n",
      "MADILLREEDLKFASTMVHALNTILLTSTELFQLRNQLKDLKTLESQNLFCCLYRSWCHN\n",
      "PVTTVSLCFLTQNYRHAYDLIQKFGDLEVTVDFLAEVDKLVQLIECPIFTYLRLQLLDVK\n",
      "NNPYLIKALYGLLMLLPQSSAFQLLSHRLQCVPNPELLQTEDSLKAAPKSQKADSPSIDY\n",
      "AELLQHFEKVQNKHLEVRHQRSGRGDHLDRRVVL\n",
      ">sp|Q5R4U3-2|TAXB1_PONAB Isoform 2 of Tax1-binding protein 1 homolog OS=Pongo abelii (Sumatran orangutan) (Pongo pygmaeus abelii). OX=['9601']\n",
      "MTSFQEVPLQTSNFAHVIFQNVAKSYLPNAHLECHYTLTPYIHPHPKDWVGIFKVGWSTA\n",
      "RDYYTFLWSPMPEHYVEGSTVNCVLAFQGYYLPNDDGEFYQFCYVTHKGEIRGASTPFQF\n",
      "RASSPVEELLTMEDEGNSDMLVVTTKAGLLELKIEKTMKEKEELLKLIAVLEKETAQLRE\n",
      "QVGRMERELNHEKERCDQLQAEQKGLTEVTQSLKMENEEFKKRFSDATSKAHQLEEDIVS\n",
      "VTHKAIEKETELDSLKDKLKKAQHEREQLECQLKTEKDEKELYKVHLKNTEIENTKLMSE\n",
      "VQTLKNLDGNKESVITHFKEEIGRLQLCLAEKENLQRTFLLTTSSKEDTFFLKEQLRKAE\n",
      "EQVQATRQEVVFLAKELSDAVNVRDRTMADLHTARLENEKVKKQLADAVAELKLNAMKKD\n",
      "QDKTDTLEHELRREVEDLKLRLQMAADHYKEKFKECQRLQKQINKLSDQSANNNNVFTKK\n",
      "MGNQQKVNDASVNTDPATSASTVDVKPSPSAAEADFDIVTKGQVCEMTKEIADKTEKYNK\n",
      "CKQLLQDEKAKCNKYADELAKMELKWKEQVKIAENVKLELAEVQDNYKELKRSLENQAER\n",
      "KMEGQNSQSPQCLKTCSEQNGYVLTLSNAQPVLQYGNPYASQETRDGADGAFYPDEIQRP\n",
      "PVRVPSWGLEDNVVCSQPARNLSRPDGLEDSEDSKEDENAPTAPDPPSQHLRGHGTGFCF\n",
      "DSSFDVHKKCPLCELMFPPNYDQSKFEEHVESHWKVCPMCSEQFPPDYDQQVFERHVQTH\n",
      "FDQNVLNFD\n",
      ">sp|P23510-2|TNFL4_HUMAN Isoform 2 of Tumor necrosis factor ligand superfamily member 4 OS=Homo sapiens (Human). OX=['9606']\n",
      "MVSHRYPRIQSIKVQFTEYKKEKGFILTSQKEDEIMKVQNNSVIINCDGFYLISLKGYFS\n",
      "QEVNISLHYQKDEEPLFQLKKVRSVNSLMVASLTYKDKVYLNVTTDNTSLDDFHVNGGEL\n",
      "ILIHQNPGEFCVL\n",
      ">sp|Q86VP1-3|TAXB1_HUMAN Isoform 3 of Tax1-binding protein 1 OS=Homo sapiens (Human). OX=['9606']\n",
      "MERELNHEKERCDQLQAEQKGLTEVTQSLKMENEEFKKRFSDATSKAHQLEEDIVSVTHK\n",
      "AIEKETELDSLKDKLKKAQHEREQLECQLKTEKDEKELYKVHLKNTEIENTKLMSEVQTL\n",
      "KNLDGNKESVITHFKEEIGRLQLCLAEKENLQRTFLLTTSSKEDTCFLKEQLRKAEEQVQ\n",
      "ATRQEVVFLAKELSDAVNVRDRTMADLHTARLENEKVKKQLADAVAELKLNAMKKDQDKT\n",
      "DTLEHELRREVEDLKLRLQMAADHYKEKFKECQRLQKQINKLSDQSANNNNVFTKKTGNQ\n",
      "QKVNDASVNTDPATSASTVDVKPSPSAAEADFDIVTKGQVCEMTKEIADKTEKYNKCKQL\n",
      "LQDEKAKCNKYADELAKMELKWKEQVKIAENVKLELAEVQDNYKELKRSLENPAERKMED\n",
      "GADGAFYPDEIQRPPVRVPSWGLEDNVVCSQPARNFSRPDGLEDSEDSKEDENVPTAPDP\n",
      "PSQHLRGHGTGFCFDSSFDVHKKCPLCELMFPPNYDQSKFEEHVESHWKVCPMCSEQFPP\n",
      "DYDQQVFERHVQTHFDQNVLNFD\n",
      ">sp|Q86VP1-4|TAXB1_HUMAN Isoform 4 of Tax1-binding protein 1 OS=Homo sapiens (Human). OX=['9606']\n",
      "MKEKEELLKLIAVLEKETAQLREQVGRMERELNHEKERCDQLQAEQKGLTEVTQSLKMEN\n",
      "EEFKKRFSDATSKAHQLEEDIVSVTHKAIEKETELDSLKDKLKKAQHEREQLECQLKTEK\n",
      "DEKELYKVHLKNTEIENTKLMSEVQTLKNLDGNKESVITHFKEEIGRLQLCLAEKENLQR\n",
      "TFLLTTSSKEDTCFLKEQLRKAEEQVQATRQEVVFLAKELSDAVNVRDRTMADLHTARLE\n",
      "NEKVKKQLADAVAELKLNAMKKDQDKTDTLEHELRREVEDLKLRLQMAADHYKEKFKECQ\n",
      "RLQKQINKLSDQSANNNNVFTKKTGNQQKVNDASVNTDPATSASTVDVKPSPSAAEADFD\n",
      "IVTKGQVCEMTKEIADKTEKYNKCKQLLQDEKAKCNKYADELAKMELKWKEQVKIAENVK\n",
      "LELAEVQDNYKELKRSLENPAERKMEDGADGAFYPDEIQRPPVRVPSWGLEDNVVCSQPA\n",
      "RNFSRPDGLEDSEDSKEDENVPTAPDPPSQHLRGHGTGFCFDSSFDVHKKCPLCELMFPP\n",
      "NYDQSKFEEHVESHWKVCPMCSEQFPPDYDQQVFERHVQTHFDQNVLNFD\n",
      ">sp|Q86VP1-2|TAXB1_HUMAN Isoform 2 of Tax1-binding protein 1 OS=Homo sapiens (Human). OX=['9606']\n",
      "MTSFQEVPLQTSNFAHVIFQNVAKSYLPNAHLECHYTLTPYIHPHPKDWVGIFKVGWSTA\n",
      "RDYYTFLWSPMPEHYVEGSTVNCVLAFQGYYLPNDDGEFYQFCYVTHKGEIRGASTPFQF\n",
      "RASSPVEELLTMEDEGNSDMLVVTTKAGLLELKIEKTMKEKEELLKLIAVLEKETAQLRE\n",
      "QVGRMERELNHEKERCDQLQAEQKGLTEVTQSLKMENEEFKKRFSDATSKAHQLEEDIVS\n",
      "VTHKAIEKETELDSLKDKLKKAQHEREQLECQLKTEKDEKELYKVHLKNTEIENTKLMSE\n",
      "VQTLKNLDGNKESVITHFKEEIGRLQLCLAEKENLQRTFLLTTSSKEDTCFLKEQLRKAE\n",
      "EQVQATRQEVVFLAKELSDAVNVRDRTMADLHTARLENEKVKKQLADAVAELKLNAMKKD\n",
      "QDKTDTLEHELRREVEDLKLRLQMAADHYKEKFKECQRLQKQINKLSDQSANNNNVFTKK\n",
      "TGNQQKVNDASVNTDPATSASTVDVKPSPSAAEADFDIVTKGQVCEMTKEIADKTEKYNK\n",
      "CKQLLQDEKAKCNKYADELAKMELKWKEQVKIAENVKLELAEVQDNYKELKRSLENPAER\n",
      "KMEDGADGAFYPDEIQRPPVRVPSWGLEDNVVCSQPARNFSRPDGLEDSEDSKEDENVPT\n",
      "APDPPSQHLRGHGTGFCFDSSFDVHKKCPLCELMFPPNYDQSKFEEHVESHWKVCPMCSE\n",
      "QFPPDYDQQVFERHVQTHFDQNVLNFD\n",
      ">sp|O35426-2|XBP1_MOUSE Isoform 2 of X-box-binding protein 1 OS=Mus musculus (Mouse). OX=['10090']\n",
      "MVVVAAAPSAATAAPKVLLLSGQPASGGRALPLMVPGPRAAGSEASGTPQARKRQRLTHL\n",
      "SPEEKALRRKLKNRVAAQTARDRKKARMSELEQQVVDLEEENHKLQLENQLLREKTHGLV\n",
      "VENQELRTRLGMDTLDPDEVPEVEAKGSGVRLVAGSAESAAGAGPVVTSPEHLPMDSDTV\n",
      "ASSDSESDILLGILDKLDPVMFFKCPSPESASLEELPEVYPEGPSSLPASLSLSVGTSSA\n",
      "KLEAINELIRFDHVYTKPLVLEIPSETESQTNVVVKIEEAPLSSSEEDHPEFIVSVKKEP\n",
      "LEDDFIPELGISNLLSSSHCLRPPSCLLDAHSDCGYEGSPSPFSDMSSPLGTDHSWEDTF\n",
      "ANELFPQLISV\n",
      ">sp|P17861-2|XBP1_HUMAN Isoform 2 of X-box-binding protein 1 OS=Homo sapiens (Human). OX=['9606']\n",
      "MVVVAAAPNPADGTPKVLLLSGQPASAAGAPAGQALPLMVPAQRGASPEAASGGLPQARK\n",
      "RQRLTHLSPEEKALRRKLKNRVAAQTARDRKKARMSELEQQVVDLEEENQKLLLENQLLR\n",
      "EKTHGLVVENQELRQRLGMDALVAEEEAEAKGNEVRPVAGSAESAAGAGPVVTPPEHLPM\n",
      "DSGGIDSSDSESDILLGILDNLDPVMFFKCPSPEPASLEELPEVYPEGPSSLPASLSLSV\n",
      "GTSSAKLEAINELIRFDHIYTKPLVLEIPSETESQANVVVKIEEAPLSPSENDHPEFIVS\n",
      "VKEEPVEDDLVPELGISNLLSSSHCPKPSSCLLDAYSDCGYGGSLSPFSDMSSLLGVNHS\n",
      "WEDTFANELFPQLISV\n",
      ">sp|Q6P132-2|TXB1B_DANRE Isoform 2 of Tax1-binding protein 1 homolog B OS=Danio rerio (Zebrafish) (Brachydanio rerio). OX=['7955']\n",
      "MALFLEGTAASSAMETSNFAHVIFQNVGKSYLPHGALECHYTLTQFIKPHPKDWVGIFKV\n",
      "GWSTARDYYTFLWSPLPDNYTEGTAINRSVVFQGYYVPNDDGEFYQFCYVTHKGEIRGAS\n",
      "TPFQFRANSPTEEELLTMEDEGGSDILVVTTKASYLEQKMEQIQQEKKELLENLDLLQKE\n",
      "RDELIDEKNRLEKEYEQERESSAQLRKDVQELQLSAQSLQEEREEVKRRMEESTARLLQL\n",
      "EEDLIGVTQKGLQKETELDCLKDRVKKLNLEKEALEGQLKNEKDEKELYKIHLKNRELEN\n",
      "TKLSAELQMLKSVDVNKENTIAQLKDELARVKSCLAEKEKQHRQLLANSSPSGESKALRE\n",
      "QLRQKEEQLQATQQQANMLKAELRDSSNARDRSMAELYRIRVEAETLKKGQADARAECSR\n",
      "LEQQLEEMKSSTQQEAQCKESDVLAVAELQREVEDLRLRLQMAAEHYKDKYKECQKLQKQ\n",
      "VVKFNEQQGVKRSPGSDAAAGPLSASPEASAPGSPSTSDAVLDAIIHGRLKSSSKELDKN\n",
      "DKYRKCKQMLNEERERCSMITDELTKMEVKLREQMKTNESLRMQLAAEEDRYKSQVAEKG\n",
      "RELKELKDSLFVLTKEKEKLEGLQKSVNREEEQKDSNLDVQSVFLQYPMPYAQDDPSPLL\n",
      "VPQRPTDLLFGNPYSSTDSRDGADGEFSDDQMPRLPPVGPPSWDSNVVCIQPARNLSRPD\n",
      "GLEEPEEPQSTQNDDEPAAPEPAEFLNDGQMPFCFEPTGEQKRCPLCEVIFPPHYDQSKF\n",
      "EEHVESHWKICPMCSEQFPLDCDQQLFEKHVLTHFDSNVLNF\n",
      ">sp|Q66HA4-2|TAXB1_RAT Isoform 2 of Tax1-binding protein 1 homolog OS=Rattus norvegicus (Rat). OX=['10116']\n",
      "MTSFQEVPLQTSNFAHVIFQNVAKSYLPNAHLECHYTLTPYIHPHSKDWVGIFKVGWSTA\n",
      "RDYYTFLWSPMPEQYVEGSTVNCVLAFQGYYLPNDDGEFYQFCYVTHKGEIRGASTPFQF\n",
      "RAASPVEELLTMEDEGNSDMLVVTTKAGLLELKIEKTLKEKEELLKLVSVLEKETAQLRE\n",
      "QVGRMERELSHEKSRCEQLQAEQKGLLEVSQSLRVENEEFMKRYSDATSKAHQLEEDIVS\n",
      "VTHKAVEKETELDSLKDKLRKAQQEKEQLECQLKTEKDEKELYKVHLKNTEIENTKLVSE\n",
      "IQTLKNVDGNKESMITHFKEEIGKLQSCLADKENLHRALLLTTSNKEDTLLLKEQLRKAE\n",
      "EQVQATRQELIFLAKELSDAVNVRDKTMADLHTARLENERVKKQLADTLAELQLHAVKTD\n",
      "QEKTDTLEHELRREVEDLKLRLQMAADHYKEKFKECQRLQKQINKLSDQSASSNGVFTKR\n",
      "MGSQQKVNDASINTDPAASASAVDVKPAASCAAETDFDMSAKDHVCEVTKEMAEKVEKYN\n",
      "KCKQLLQDEKAKCNKYADELAQMELKWKEQVRIAENVKLELAELEDSYSVSSENVRLELA\n",
      "ELEDSYSELTKSLEDQKGRKMEGQSPQQVSRCLNTCSEQSGLLPTLPAAQPVLQYGNPYT\n",
      "AHETRDGADGAFYPDEIQRPPVRGPSWEDNVVCSQPARNLSRPDGLEDPEDSREDENVPI\n",
      "PPDPANQHLRGHGAGFCFDSSFDVHKKCPLCELMFPPNYDQIKFEEHVESHWKLPGEFFI\n",
      "EYPWATYAPHETLGPWRRLGKVLAVLSSDTQLWCEKPDISLGRQKQKILGMCWSESRQIS\n",
      "DLQVQ\n",
      ">sp|Q9UQ35-3|SRRM2_HUMAN Isoform 3 of Serine/arginine repetitive matrix protein 2 OS=Homo sapiens (Human). OX=['9606']\n",
      "MLLEKDVNPGGKEETPGQRPAVTETHQLAELNEKKNERLRAAFGISDSYVDGSSFDPQRR\n",
      "AREAKQPAPEPPKPYSLVRESSSSRSPTPKQKKKKKKKDRGRRSESSSPRRERKKSSKKK\n",
      "KHRSESESKKRKHRSPTPKSKRKSKDKKRKRSRSTTPAPKSRRAHRSTSADSASSSDTSR\n",
      "SRSRSAAAKTHTTALAGRSPSPASGRRGEGDAPFSEPGTTSTQRPSSPETATKQPSSPYE\n",
      "DKDKDKKEKSATRPSPSPERSSTGPEPPAPTPLLAERHGGSPQPLATTPLSQEPVNPSPQ\n",
      "LFMFLKGILVF\n",
      ">sp|Q9UQ35-2|SRRM2_HUMAN Isoform 2 of Serine/arginine repetitive matrix protein 2 OS=Homo sapiens (Human). OX=['9606']\n",
      "MYNGIGLPTPRGSGTNGYVQRNLSLVRGRRGERPDYKGEEELRRLEAALVKRPNPDILDH\n",
      "ERKRRVELRCLELEEMMEEQGYEEQQIQEKVATFRLMLLEKDVNPGGKEETPGQRPAVTE\n",
      "THQLAELNEKKNERLRAAFGISDSYVDGSSFDPQRRAREAKQPAPEPPKPYSLVRESSSS\n",
      "RSPTPKQKKKKKKKDRGRRSESSSPRRERKKSSKKKKHRSESESKKRKHRSPTPKSKRKS\n",
      "KDKKRKRSRSTTPAPKSRRAHRSTSADSASSSDTSRSRSRSAAAKTHTTALAGRSPSPAS\n",
      "GRRGEGDAPFSEPGTTSTQRPSSPETATKQPSSPYEDKDKDKKEKSATRPSPSPERSSTG\n",
      "PEPPAPTPLLAERHGGSPQPLATTPLSQEPVNPPSEASPTRDRSPPKSPEKLPQSSSSES\n",
      "SPPSPQPTKVSRHASSSPESPKPAPAPGSHREISSSPTSKNRSHGRAKRDKSHSHTPSRR\n",
      "MGRSRSPATAKRGRSRSRTPTKRGHSRSRSPQWRRSRSAQRWGRSRSPQRRGRSRSPQRP\n",
      "GWSRSRNTQRRGRSRSARRGRSHSRSPATRGRSRSRTPARRGRSRSRTPARRRSRSRTPT\n",
      "RRRSRSRTPARRGRSRSRTPARRRSRTRSPVRRRSRSRSPARRSGRSRSRTPARRGRSRS\n",
      "RTPARRGRSRSRTPARRSGRSRSRTPARRGRSRSRTPRRGRSRSRSLVRRGRSHSRTPQR\n",
      "RGRSGSSSERKNKSRTSQRRSRSNSSPEMKKSRISSRRSRSLSSPRSKAKSRLSLRRSLS\n",
      "GSSPCPKQKSQTPPRRSRSGSSQPKAKSRTPPRRSRSSSSPPPKQKSKTPSRQSHSSSSP\n",
      "HPKVKSGTPPRQGSITSPQANEQSVTPQRRSCFESSPDPELKSRTPSRHSCSGSSPPRVK\n",
      "SSTPPRQSPSRSSSPQPKVKAIISPRQRSHSGSSSPSPSRVTSRTTPRRSRSVSPCSNVE\n",
      "SRLLPRYSHSGSSSPDTKVKPETPPRQSHSGSISPYPKVKAQTPPGPSLSGSKSPCPQEK\n",
      "SKDSLVQSCPGSLSLCAGVKSSTPPGESYFGVSSLQLKGQSQTSPDHRSDTSSPEVRQSH\n",
      "SESPSLQSKSQTSPKGGRSRSSSPVTELASRSPIRQDRGEFSASPMLKSGMSPEQSRFQS\n",
      "DSSSYPTVDSNSLLGQSRLETAESKEKMALPPQEDATASPPRQKDKFSPFPVQDRPESSL\n",
      "VFKDTLRTPPRERSGAGSSPETKEQNSALPTSSQDEELMEVVEKSEEPAGQILSHLSSEL\n",
      "KEMSTSNFESSPEVEERPAVSLTLDQSQSQASLEAVEVPSMASSWGGPHFSPEHKELSNS\n",
      "PLRENSFGSPLEFRNSGPLGTEMNTGFSSEVKEDLNGPFLNQLETDPSLDMKEQSTRSSG\n",
      "HSSSELSPDAVEKAGMSSNQSISSPVLDAVPRTPSRERSSSASSPEMKDGLPRTPSRRSR\n",
      "SGSSPGLRDGSGTPSRHSLSGSSPGMKDIPRTPSRGRSECDSSPEPKALPQTPRPRSRSP\n",
      "SSPELNNKCLTPQRERSGSESSVDQKTVARTPLGQRSRSGSSQELDVKPSASPQERSESD\n",
      "SSPDSKAKTRTPLRQRSRSGSSPEVDSKSRLSPRRSRSGSSPEVKDKPRAAPRAQSGSDS\n",
      "SPEPKAPAPRALPRRSRSGSSSKGRGPSPEGSSSTESSPEHPPKSRTARRGSRSSPEPKT\n",
      "KSRTPPRRRSSRSSPELTRKARLSRRSRSASSSPETRSRTPPRHRRSPSVSSPEPAEKSR\n",
      "SSRRRRSASSPRTKTTSRRGRSPSPKPRGLQRSRSRSRREKTRTTRRRDRSGSSQSTSRR\n",
      "RQRSRSRSRVTRRRRGGSGYHSRSPARQESSRTSSRRRRGRSRTPPTSRKRSRSRTSPAP\n",
      "WKRSRSRASPATHRRSRSRTPLISRRRSRSRTSPVSRRRSRSRTSVTRRRSRSRASPVSR\n",
      "RRSRSRTPPVTRRRSRSRTPTTRRRSRSRTPPVTRRRSRSRTPPVTRRRSRSRTSPITRR\n",
      "RSRSRTSPVTRRRSRSRTSPVTRRRSRSRTSPVTRRRSRSRTPPAQDQPRSPVPSAFSDQ\n",
      "SRCLIAQTTPVAGSQSLSSGAVATTTSSAGDHNGMLSVPAPGVPHSDVGEPPASTGAQQP\n",
      "SALAALQPAKERRSSSSSSSSSSSSSSSSSSSSSSSSSGSSSSDSEGSSLPVQPEVALKR\n",
      "VPSPTPAPKEAVREGRPPEPTPAKRKRRSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS\n",
      "SSSSSSSSSSPSPAKPGPQALPKPASPKKPPPGERRSRSPRKPIDSLRDSRSLSYSPVER\n",
      "RRPSPQPSPRDQQSSSSERGSRRGQRGDSRSPSHKRRRETPSPRPMRHRSSRSP\n",
      ">sp|Q13884-2|SNTB1_HUMAN Isoform 2 of Beta-1-syntrophin OS=Homo sapiens (Human). OX=['9606']\n",
      "MAVAAAAAAAGPAGAGGGRAQRSGLLEVLVRDRWHKVLVNLSEDALVLSSEEGAAAYNGI\n",
      "GTATNGSFCRGAGAGHPGAGGAQPPDSPAGVRTAFTDLPEQVPESISNQKRGVKVLKQEL\n",
      "GGLGISIKGGKENKMPILISKIFKGLAADQTQALYVGDAILSVNGADLRDATHDEAVQAL\n",
      "KRAGKEVLLEVKYMREATPYVKKGSPVSEIGWETPPPESPRLGGSTSDPPSSQSFSFHRD\n",
      "RKSIPLKMCYVTRSMALADPENRQLEIHSPDAKHTVILRSKDSATAQAWFSAIHSNVNDL\n",
      "LTRVIAEVREQLGKTGIAGSREIRHLGWLAEKVPGESKKQWKPALVVLTEKDLLIYDSMP\n",
      "RRKEAWFSPVHTYPLLATSPHP\n",
      ">sp|Q9NPB6-2|PAR6A_HUMAN Isoform 2 of Partitioning defective 6 homolog alpha OS=Homo sapiens (Human). OX=['9606']\n",
      "MARPQRTPARSPDSIVEVKSKFDAEFRRFALPRASVSGFQEFSRLLRAVHQIPGLDVLLG\n",
      "YTDAHGDLLPLTNDDSLHRALASGPPPLRLLVQKREADSSGLAFASNSLQRRKKGLLLRP\n",
      "VAPLRTRPPLLISLPQDFRQVSSVIDVDLLPETHRRVRLHKHGSDRPLGFYIRDGMSVRV\n",
      "APQGLERVPGIFISRLVRGGLAESTGLLAVSDEILEVNGIEVAGKTLDQVTDMMVANSHN\n",
      "LIVTVKPANQRNNVVRGASGRLTGPPSAGPGPAEPDSDDDSSDLVIENRQPPSSNGLSQG\n",
      "PPCWDLHPGCRHPGTRSSLPSLDDQEQASSGWGSRIRGDGSGFSL\n"
     ]
    }
   ],
   "source": [
    "for e in entries:\n",
    "    if 'Alternative splicing' in e.keywords:\n",
    "        for i in e.isoforms():\n",
    "            print(i)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We would like to run a FASTA similarity search against Swiss-Prot for one of the sequences. Let's take the canonical sequence of the first entry in entries.\n",
    "\n",
    "Here we use the ebiwebservices module from prunito."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unlike searching e.g. UniProt, which uses a GET request behind the scenes, running a sequence similarity search is usually a bit more work:\n",
    "\n",
    "Our code | Web service | Verb\n",
    "--- | --- | ---\n",
    "Prepare a request: seq, db, parameters | | \n",
    "Send request | | POST\n",
    "- | Start search | -\n",
    "- | Send job ID | -\n",
    "Save job ID | - | -\n",
    "Ask whether <job ID> done | - | GET\n",
    "repeat | |\n",
    "Retrieve results |- | GET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from prunito import ebiwebservices as ews"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp|Q06507|ATF4_MOUSE Cyclic AMP-dependent transcription factor ATF-4 OS=Mus musculus (Mouse). OX=['10090']\n",
      "MTEMSFLNSEVLAGDLMSPFDQSGLGAEESLGLLDDYLEVAKHLKPHGFSSDKAGSSEWP\n",
      "AMDDGLASASDTGKEDAFSGTDWMLEKMDLKEFDFDALFRMDDLETMPDELLTTLDDTCD\n",
      "LFAPLVQETNKEPPQTVNPIGHLPESLIKVDQVAPFTFLQPFPCSPGVLSSTPEHSFSLE\n",
      "LGSEVDISEGDRKPDSAAYITLIPPCVKEEDTPSDNDSGICMSPESYLGSPQHSPSTSRA\n",
      "PPDNLPSPGGSRGSPRPKPYDPPGVSLTAKVKTEKLDKKLKKMEQNKTAATRYRQKKRAE\n",
      "QEALTGECKELEKKNEALKEKADSLAKEIQYLKDLIEEVRKARGKKRVP\n"
     ]
    }
   ],
   "source": [
    "first_entry = entries[0]\n",
    "print(first_entry.as_fasta())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No email address for EBI web service set.\n",
      "This is required for the services to work.\n",
      "Use set_email() to provide an email address.\n"
     ]
    }
   ],
   "source": [
    "# fasta_search() defaults to Swiss-Prot as target set\n",
    "similar = ews.fasta_search(first_entry.as_fasta())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Aha, we get a message that EBI web services require an email address. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "ews.set_email('some@gmx.de')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on FASTASimilaritySearch in module prunito.ebiwebservices object:\n",
      "\n",
      "class FASTASimilaritySearch(EBIWebService)\n",
      " |  Use FASTA suite to search a protein sequence against a database.\n",
      " |  \n",
      " |  The suite contains the different programs - fasta, fastx, fasty, ssearch,\n",
      " |  ggsearch, glsearch, tfastx, tfasty.\n",
      " |  \n",
      " |  fasta: Scan a protein or DNA sequence library for similar sequences.\n",
      " |      protein:protein or DNA:DNA\n",
      " |  fastx: Compares a DNA sequence to a protein sequence database, translating\n",
      " |      the DNA sequence in three forward (or reverse) frames and allowing\n",
      " |      frameshifts.\n",
      " |      DNA:protein\n",
      " |  fasty: Compares a DNA sequence to a protein sequence database, translating\n",
      " |      the DNA sequence in three forward (or reverse) frames and allowing\n",
      " |      frameshifts.\n",
      " |      DNA:protein\n",
      " |  ssearch: Compare a protein or DNA sequence to a sequence database using\n",
      " |      the Smith-Waterman algorithm.\n",
      " |      protein:protein or DNA:DNA\n",
      " |  ggsearch: Compare a protein or DNA sequence to a sequence database using\n",
      " |      a global alignment (Needleman-Wunsch).\n",
      " |      protein:protein or DNA:DNA\n",
      " |  glsearch: Compare a protein or DNA sequence to a sequence database with\n",
      " |      alignments that are global in the query and local in the database\n",
      " |      sequence (global-local).\n",
      " |      protein:protein or DNA:DNA\n",
      " |  tfastx: Compares a protein sequence to a DNA sequence or DNA sequence\n",
      " |      library. The DNA sequence is translated in three forward and three\n",
      " |      reverse frames, and the protein query sequence is compared to each\n",
      " |      of the six derived protein sequences. The DNA sequence is translated\n",
      " |      from one end to the other; no attempt is made to edit out intervening\n",
      " |      sequences. Termination codons are translated into unknown ('X') amino\n",
      " |      acids.\n",
      " |  \n",
      " |  Method resolution order:\n",
      " |      FASTASimilaritySearch\n",
      " |      EBIWebService\n",
      " |      builtins.object\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __call__(self, seq, program='fasta', database='uniprotkb_swissprot', stype='protein', **kwargs)\n",
      " |      Run a sequence similarity search using FASTA suite tools.\n",
      " |      \n",
      " |      The FASTA tools take many more arguments than listed below which are only the\n",
      " |      mandatory ones. Any further keyword arguments passed to the __call__() method\n",
      " |      will be added to the parameters of the FASTA run.\n",
      " |      \n",
      " |      Args:\n",
      " |          seq (str): Query sequence(s); preferably in FASTA format.\n",
      " |          database (str): Target database to search against. Default: uniprotkb_swissprot.\n",
      " |              Many choices like uniprotkb, uniprotkb_swissprot, uniprotkb_trembl. For a\n",
      " |              full list run get_target_databases().\n",
      " |          stype (str): Type of query sequence. Default: protein.\n",
      " |              Can be: protein, dna, rna.\n",
      " |      \n",
      " |      Returns:\n",
      " |          WSResponse\n",
      " |  \n",
      " |  __init__(self)\n",
      " |      Initialize self.  See help(type(self)) for accurate signature.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from EBIWebService:\n",
      " |  \n",
      " |  get_parameter_details(self, param)\n",
      " |      Get details for parameter.\n",
      " |      \n",
      " |      Details include the name, description, type and possible values.\n",
      " |      \n",
      " |      Args:\n",
      " |          param (str): A parameter for the web service.\n",
      " |      \n",
      " |      Returns:\n",
      " |          dict: Parameter details.\n",
      " |  \n",
      " |  get_parameters(self)\n",
      " |      Retrieve a list of all parameters for the web service.\n",
      " |      \n",
      " |      To learn more about possible values for any given parameter,\n",
      " |      use the get_parameter_details() method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          list of str: Parameters.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from EBIWebService:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes inherited from EBIWebService:\n",
      " |  \n",
      " |  email = 'some@gmx.de'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(ews.fasta_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['program',\n",
       " 'stype',\n",
       " 'matrix',\n",
       " 'match_scores',\n",
       " 'gapopen',\n",
       " 'gapext',\n",
       " 'hsps',\n",
       " 'expupperlim',\n",
       " 'explowlim',\n",
       " 'strand',\n",
       " 'hist',\n",
       " 'scores',\n",
       " 'alignments',\n",
       " 'scoreformat',\n",
       " 'stats',\n",
       " 'annotfeats',\n",
       " 'annotsym',\n",
       " 'dbrange',\n",
       " 'seqrange',\n",
       " 'filter',\n",
       " 'transltable',\n",
       " 'sequence',\n",
       " 'database',\n",
       " 'ktup']"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ews.fasta_search.get_parameters()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'description': 'The databases to run the sequence similarity search against. Multiple databases can be used at the same time',\n",
       " 'name': 'Database',\n",
       " 'type': 'STRING',\n",
       " 'values': [('UniProt Knowledgebase (The UniProt Knowledgebase includes UniProtKB/Swiss-Prot and UniProtKB/TrEMBL)',\n",
       "   'uniprotkb'),\n",
       "  ('UniProtKB/Swiss-Prot (The manually annotated section of UniProtKB)',\n",
       "   'uniprotkb_swissprot'),\n",
       "  ('UniProtKB/Swiss-Prot isoforms (The manually annotated isoforms of UniProtKB/Swiss-Prot)',\n",
       "   'uniprotkb_swissprotsv'),\n",
       "  ('UniProtKB/TrEMBL (The automatically annotated section of UniProtKB)',\n",
       "   'uniprotkb_trembl'),\n",
       "  ('UniProtKB Archaea', 'uniprotkb_archaea'),\n",
       "  ('UniProtKB Arthropoda', 'uniprotkb_arthropoda'),\n",
       "  ('UniProtKB Bacteria', 'uniprotkb_bacteria'),\n",
       "  ('UniProtKB Complete Microbial Proteomes',\n",
       "   'uniprotkb_complete_microbial_proteomes'),\n",
       "  ('UniProtKB Eukaryota', 'uniprotkb_eukaryota'),\n",
       "  ('UniProtKB Fungi', 'uniprotkb_fungi'),\n",
       "  ('UniProtKB Human', 'uniprotkb_human'),\n",
       "  ('UniProtKB Mammals', 'uniprotkb_mammals'),\n",
       "  ('UniProtKB Nematoda', 'uniprotkb_nematoda'),\n",
       "  ('UniProtKB PDB', 'uniprotkb_pdb'),\n",
       "  ('UniProtKB Rodents', 'uniprotkb_rodents'),\n",
       "  ('UniProtKB Vertebrates', 'uniprotkb_vertebrates'),\n",
       "  ('UniProtKB Viridiplantae', 'uniprotkb_viridiplantae'),\n",
       "  ('UniProtKB Viruses', 'uniprotkb_viruses'),\n",
       "  ('UniProt Clusters 100% (UniRef100)', 'uniref100'),\n",
       "  ('UniProt Clusters 100% (SEG filtered version of UniRef100)',\n",
       "   'uniref100_seg'),\n",
       "  ('UniProt Clusters 90% (UniRef90)', 'uniref90'),\n",
       "  ('UniProt Clusters 50% (UniRef50)', 'uniref50'),\n",
       "  ('EPO Patent Protein Sequences', 'epop'),\n",
       "  ('JPO Patent Protein Sequences', 'jpop'),\n",
       "  ('KIPO Patent Protein Sequences', 'kpop'),\n",
       "  ('USPTO Patent Protein Sequences', 'uspop'),\n",
       "  ('NR Patent Proteins Level-1', 'nrpl1'),\n",
       "  ('NR Patent Proteins Level-2', 'nrpl2'),\n",
       "  ('Protein Structure Sequences (PDBe protein structure sequences)', 'pdb'),\n",
       "  ('Structural Genomics Targets (PSI protein structure sequences)', 'sgt'),\n",
       "  ('UniProt Archive (Sequences from the UniProt Archive UniParc)', 'uniparc'),\n",
       "  ('IntAct (Sequences from IntAct interactors)', 'intact'),\n",
       "  ('IDP-IMGT/HLA (Immuno Polymorphism Database-International Immunogenetics project/Human Leucocyte Antigen)',\n",
       "   'imgthlap'),\n",
       "  ('IPD-KIR (Immuno Polymorphism Database-Killer-cell Immunoglobulin-like Receptors)',\n",
       "   'ipdkirp'),\n",
       "  ('IPD-MHC (Immuno Polymorphism Database-Major Histocompatibility Complex)',\n",
       "   'ipdmhcp'),\n",
       "  ('MACiE Annot Pub', 'macie_annot.pub'),\n",
       "  ('MEROPS-MPRO (Sequences from the MEROPS scan dataset)', 'mpro'),\n",
       "  ('MEROPS-MPEP (Sequences from the peptidase or inhibitor domain sequence only)',\n",
       "   'mpep'),\n",
       "  ('MEROPS-MP (Sequences from the full MEROPS collection)', 'mp'),\n",
       "  ('ENA Sequence Release', 'em_rel'),\n",
       "  ('ENA Sequence Environmental', 'em_rel_env'),\n",
       "  ('ENA Sequence EST Environmental', 'em_rel_est_env'),\n",
       "  ('ENA Sequence GSS Environmental', 'em_rel_gss_env'),\n",
       "  ('ENA Sequence HTC Environmental', 'em_rel_htc_env'),\n",
       "  ('ENA Sequence HTG Environmental', 'em_rel_htg_env'),\n",
       "  ('ENA Sequence Patent Environmental', 'em_rel_pat_env'),\n",
       "  ('ENA Sequence Standard Environmental', 'em_rel_std_env'),\n",
       "  ('ENA Sequence STS Environmental', 'em_rel_sts_env'),\n",
       "  ('ENA Sequence TSA Environmental', 'em_rel_tsa_env'),\n",
       "  ('ENA Sequence Fungi', 'em_rel_fun'),\n",
       "  ('ENA Sequence EST Fungi', 'em_rel_est_fun'),\n",
       "  ('ENA Sequence GSS Fungi', 'em_rel_gss_fun'),\n",
       "  ('ENA Sequence HTC Fungi', 'em_rel_htc_fun'),\n",
       "  ('ENA Sequence HTG Fungi', 'em_rel_htg_fun'),\n",
       "  ('ENA Sequence Patent Fungi', 'em_rel_pat_fun'),\n",
       "  ('ENA Sequence Standard Fungi', 'em_rel_std_fun'),\n",
       "  ('ENA Sequence STS Fungi', 'em_rel_sts_fun'),\n",
       "  ('ENA Sequence TSA Fungi', 'em_rel_tsa_fun'),\n",
       "  ('ENA Sequence Human', 'em_rel_hum'),\n",
       "  ('ENA Sequence EST Human', 'em_rel_est_hum'),\n",
       "  ('ENA Sequence GSS Human', 'em_rel_gss_hum'),\n",
       "  ('ENA Sequence HTC Human', 'em_rel_htc_hum'),\n",
       "  ('ENA Sequence HTG Human', 'em_rel_htg_hum'),\n",
       "  ('ENA Sequence Patent Human', 'em_rel_pat_hum'),\n",
       "  ('ENA Sequence Standard Human', 'em_rel_std_hum'),\n",
       "  ('ENA Sequence STS Human', 'em_rel_sts_hum'),\n",
       "  ('ENA Sequence Invertebrate', 'em_rel_inv'),\n",
       "  ('ENA Sequence EST Invertebrate', 'em_rel_est_inv'),\n",
       "  ('ENA Sequence GSS Invertebrate', 'em_rel_gss_inv'),\n",
       "  ('ENA Sequence HTC Invertebrate', 'em_rel_htc_inv'),\n",
       "  ('ENA Sequence HTG Invertebrate', 'em_rel_htg_inv'),\n",
       "  ('ENA Sequence Patent Invertebrate', 'em_rel_pat_inv'),\n",
       "  ('ENA Sequence Standard Invertebrate', 'em_rel_std_inv'),\n",
       "  ('ENA Sequence STS Invertebrate', 'em_rel_sts_inv'),\n",
       "  ('ENA Sequence TSA Invertebrate', 'em_rel_tsa_inv'),\n",
       "  ('ENA Sequence Mammal', 'em_rel_mam'),\n",
       "  ('ENA Sequence EST Mammal', 'em_rel_est_mam'),\n",
       "  ('ENA Sequence GSS Mammal', 'em_rel_gss_mam'),\n",
       "  ('ENA Sequence HTC Mammal', 'em_rel_htc_mam'),\n",
       "  ('ENA Sequence HTG Mammal', 'em_rel_htg_mam'),\n",
       "  ('ENA Sequence Patent Mammal', 'em_rel_pat_mam'),\n",
       "  ('ENA Sequence Standard Mammal', 'em_rel_std_mam'),\n",
       "  ('ENA Sequence STS Mammal', 'em_rel_sts_mam'),\n",
       "  ('ENA Sequence TSA Mammal', 'em_rel_tsa_mam'),\n",
       "  ('ENA Sequence Mouse', 'em_rel_mus'),\n",
       "  ('ENA Sequence EST Mouse', 'em_rel_est_mus'),\n",
       "  ('ENA Sequence GSS Mouse', 'em_rel_gss_mus'),\n",
       "  ('ENA Sequence HTC Mouse', 'em_rel_htc_mus'),\n",
       "  ('ENA Sequence HTG Mouse', 'em_rel_htg_mus'),\n",
       "  ('ENA Sequence Patent Mouse', 'em_rel_pat_mus'),\n",
       "  ('ENA Sequence Standard Mouse', 'em_rel_std_mus'),\n",
       "  ('ENA Sequence STS Mouse', 'em_rel_sts_mus'),\n",
       "  ('ENA Sequence Phage', 'em_rel_phg'),\n",
       "  ('ENA Sequence GSS Phage', 'em_rel_gss_phg'),\n",
       "  ('ENA Sequence HTG Phage', 'em_rel_htg_phg'),\n",
       "  ('ENA Sequence Patent Phage', 'em_rel_pat_phg'),\n",
       "  ('ENA Sequence Standard Phage', 'em_rel_std_phg'),\n",
       "  ('ENA Sequence Plant', 'em_rel_pln'),\n",
       "  ('ENA Sequence EST Plant', 'em_rel_est_pln'),\n",
       "  ('ENA Sequence GSS Plant', 'em_rel_gss_pln'),\n",
       "  ('ENA Sequence HTC Plant', 'em_rel_htc_pln'),\n",
       "  ('ENA Sequence HTG Plant', 'em_rel_htg_pln'),\n",
       "  ('ENA Sequence Patent Plant', 'em_rel_pat_pln'),\n",
       "  ('ENA Sequence Standard Plant', 'em_rel_std_pln'),\n",
       "  ('ENA Sequence STS Plant', 'em_rel_sts_pln'),\n",
       "  ('ENA Sequence TSA Plant', 'em_rel_tsa_pln'),\n",
       "  ('ENA Sequence Prokaryote', 'em_rel_pro'),\n",
       "  ('ENA Sequence EST Prokaryote', 'em_rel_est_pro'),\n",
       "  ('ENA Sequence GSS Prokaryote', 'em_rel_gss_pro'),\n",
       "  ('ENA Sequence HTC Prokaryote', 'em_rel_htc_pro'),\n",
       "  ('ENA Sequence HTG Prokaryote', 'em_rel_htg_pro'),\n",
       "  ('ENA Sequence Patent Prokaryote', 'em_rel_pat_pro'),\n",
       "  ('ENA Sequence Standard Prokaryote', 'em_rel_std_pro'),\n",
       "  ('ENA Sequence STS Prokaryote', 'em_rel_sts_pro'),\n",
       "  ('ENA Sequence TSA Prokaryote', 'em_rel_tsa_pro'),\n",
       "  ('ENA Sequence Rodent', 'em_rel_rod'),\n",
       "  ('ENA Sequence EST Rodent', 'em_rel_est_rod'),\n",
       "  ('ENA Sequence GSS Rodent', 'em_rel_gss_rod'),\n",
       "  ('ENA Sequence HTC Rodent', 'em_rel_htc_rod'),\n",
       "  ('ENA Sequence HTG Rodent', 'em_rel_htg_rod'),\n",
       "  ('ENA Sequence Patent Rodent', 'em_rel_pat_rod'),\n",
       "  ('ENA Sequence Standard Rodent', 'em_rel_std_rod'),\n",
       "  ('ENA Sequence STS Rodent', 'em_rel_sts_rod'),\n",
       "  ('ENA Sequence TSA Rodent', 'em_rel_tsa_rod'),\n",
       "  ('ENA Sequence Synthetic', 'em_rel_syn'),\n",
       "  ('ENA Sequence Patent Synthetic', 'em_rel_pat_syn'),\n",
       "  ('ENA Sequence Standard Synthetic', 'em_rel_std_syn'),\n",
       "  ('ENA Sequence Transgenic', 'em_rel_tgn'),\n",
       "  ('ENA Sequence Standard Transgenic', 'em_rel_std_tgn'),\n",
       "  ('ENA Sequence GSS Transgenic', 'em_rel_gss_tgn'),\n",
       "  ('ENA Sequence Unclassified', 'em_rel_unc'),\n",
       "  ('ENA Sequence EST Unclassified', 'em_rel_est_unc'),\n",
       "  ('ENA Sequence Patent Unclassified', 'em_rel_pat_unc'),\n",
       "  ('ENA Sequence Standard Unclassified', 'em_rel_std_unc'),\n",
       "  ('ENA Sequence Viral', 'em_rel_vrl'),\n",
       "  ('ENA Sequence EST Viral', 'em_rel_est_vrl'),\n",
       "  ('ENA Sequence GSS Viral', 'em_rel_gss_vrl'),\n",
       "  ('ENA Sequence HTG Viral', 'em_rel_htg_vrl'),\n",
       "  ('ENA Sequence Patent Viral', 'em_rel_pat_vrl'),\n",
       "  ('ENA Sequence Standard Viral', 'em_rel_std_vrl'),\n",
       "  ('ENA Sequence TSA Viral', 'em_rel_tsa_vrl'),\n",
       "  ('ENA Sequence Vertebrate', 'em_rel_vrt'),\n",
       "  ('ENA Sequence EST Vertebrate', 'em_rel_est_vrt'),\n",
       "  ('ENA Sequence GSS Vertebrate', 'em_rel_gss_vrt'),\n",
       "  ('ENA Sequence HTC Vertebrate', 'em_rel_htc_vrt'),\n",
       "  ('ENA Sequence HTG Vertebrate', 'em_rel_htg_vrt'),\n",
       "  ('ENA Sequence Patent Vertebrate', 'em_rel_pat_vrt'),\n",
       "  ('ENA Sequence Standard Vertebrate', 'em_rel_std_vrt'),\n",
       "  ('ENA Sequence STS Vertebrate', 'em_rel_sts_vrt'),\n",
       "  ('ENA Sequence TSA Vertebrate', 'em_rel_tsa_vrt'),\n",
       "  ('ENA Sequence Updates', 'emnew'),\n",
       "  ('ENA Coding Sequence Release', 'em_cds_rel'),\n",
       "  ('ENA Coding Sequence Updates', 'em_cds_cum'),\n",
       "  ('ENA Non-Coding Sequence Release', 'em_ncr_rel'),\n",
       "  ('ENA Non-Coding Sequence Updates', 'em_ncr_cum'),\n",
       "  ('ENA Sequence Expressed Sequence Tag', 'em_rel_est'),\n",
       "  ('ENA Sequence Genome Survey Sequence', 'em_rel_gss'),\n",
       "  ('ENA Sequence High Throughput cDNA', 'em_rel_htc'),\n",
       "  ('ENA Sequence High Throughput Genome', 'em_rel_htg'),\n",
       "  ('ENA Sequence Patent', 'em_rel_pat'),\n",
       "  ('ENA Sequence Standard', 'em_rel_std'),\n",
       "  ('ENA Sequence Sequence Tagged Site', 'em_rel_sts'),\n",
       "  ('ENA Sequence Transcriptome Shotgun Assembly', 'em_rel_tsa'),\n",
       "  ('ENA Sequence Release and Updates', 'emall'),\n",
       "  ('ENA Sequence Vectors', 'emvec'),\n",
       "  ('IMGT/LIGM-DB', 'imgtligm'),\n",
       "  ('IMGT/HLA', 'imgthla'),\n",
       "  ('IPD-KIR', 'ipdkir'),\n",
       "  ('IPD-MHC', 'ipdmhc'),\n",
       "  ('NR Patent DNAs Level-1', 'nrnl1'),\n",
       "  ('NR Patent DNAs Level-2', 'nrnl2'),\n",
       "  ('Nucleotide Structure Sequences', 'pdbna'),\n",
       "  ('ENA Barcode Sequence Release', 'em_bcd_rel'),\n",
       "  ('ENA Barcode Sequence Release Fungi', 'em_bcd_rel_fun'),\n",
       "  ('ENA Barcode Sequence Release Invertebrate', 'em_bcd_rel_inv'),\n",
       "  ('ENA Barcode Sequence Release Mammal', 'em_bcd_rel_mam'),\n",
       "  ('ENA Barcode Sequence Release Mouse', 'em_bcd_rel_mus'),\n",
       "  ('ENA Barcode Sequence Release Plant', 'em_bcd_rel_pln'),\n",
       "  ('ENA Barcode Sequence Release Prokaryote', 'em_bcd_rel_pro'),\n",
       "  ('ENA Barcode Sequence Release Rodent', 'em_bcd_rel_rod'),\n",
       "  ('ENA Barcode Sequence Release Unclassified', 'em_bcd_rel_unc'),\n",
       "  ('ENA Barcode Sequence Release Vertebrate', 'em_bcd_rel_vrt'),\n",
       "  ('ENA Barcode Sequence Release Standard', 'em_bcd_rel_std'),\n",
       "  ('ENA Barcode Sequence Updates', 'em_bcd_cum'),\n",
       "  ('ENA Geospatial Release Environment', 'em_geo_rel_env'),\n",
       "  ('ENA Geospatial Release Fungi', 'em_geo_rel_fun'),\n",
       "  ('ENA Geospatial Release Human', 'em_geo_rel_hum'),\n",
       "  ('ENA Geospatial Release Invertebrate', 'em_geo_rel_inv'),\n",
       "  ('ENA Geospatial Release Mammal', 'em_geo_rel_mam'),\n",
       "  ('ENA Geospatial Release Contigs', 'em_geo_rel_phg'),\n",
       "  ('ENA Geospatial Release Plant', 'em_geo_rel_pln'),\n",
       "  ('ENA Geospatial Release Prokaryote', 'em_geo_rel_pro'),\n",
       "  ('ENA Geospatial Release Rodent', 'em_geo_rel_rod'),\n",
       "  ('ENA Geospatial Release Viral', 'em_geo_rel_vrl'),\n",
       "  ('ENA Geospatial Release Vertebrate', 'em_geo_rel_vrt'),\n",
       "  ('ENA Geospatial Release Contigs', 'em_geo_rel_con'),\n",
       "  ('ENA Geospatial Release Standard', 'em_geo_rel_std'),\n",
       "  ('ENA Geospatial Release TSA', 'em_geo_rel_tsa'),\n",
       "  ('ENA Geospatial Release WGS', 'em_geo_rel_wgs'),\n",
       "  ('ENA Contigs Release Prokaryote', 'em_rel_con_pro'),\n",
       "  ('ENA Contigs Updates Prokaryote', 'em_cum_con_pro'),\n",
       "  ('ENA Coding Sequence Release Contigs', 'em_cds_rel_con'),\n",
       "  ('ENA Coding Sequence Release EST', 'em_cds_rel_est'),\n",
       "  ('ENA Coding Sequence Release GSS', 'em_cds_rel_gss'),\n",
       "  ('ENA Coding Sequence Release HTC', 'em_cds_rel_htc'),\n",
       "  ('ENA Coding Sequence Release HTG', 'em_cds_rel_htg'),\n",
       "  ('ENA Coding Sequence Release Patent', 'em_cds_rel_pat'),\n",
       "  ('ENA Coding Sequence Release Standard', 'em_cds_rel_std'),\n",
       "  ('ENA Coding Sequence Release STS', 'em_cds_rel_sts'),\n",
       "  ('ENA Coding Sequence Release TSA', 'em_cds_rel_tsa'),\n",
       "  ('ENA Coding Sequence Release WGS', 'em_cds_rel_wgs'),\n",
       "  ('ENA Coding Sequence Release Environment', 'em_cds_rel_env'),\n",
       "  ('ENA Coding Sequence Release Fungi', 'em_cds_rel_fun'),\n",
       "  ('ENA Coding Sequence Release Human', 'em_cds_rel_hum'),\n",
       "  ('ENA Coding Sequence Release Invertebrate', 'em_cds_rel_inv'),\n",
       "  ('ENA Coding Sequence Release Mammal', 'em_cds_rel_mam'),\n",
       "  ('ENA Coding Sequence Release Mouse', 'em_cds_rel_mus'),\n",
       "  ('ENA Coding Sequence Release Phage', 'em_cds_rel_phg'),\n",
       "  ('ENA Coding Sequence Release Plant', 'em_cds_rel_pln'),\n",
       "  ('ENA Coding Sequence Release Prokaryote', 'em_cds_rel_pro'),\n",
       "  ('ENA Coding Sequence Release Rodent', 'em_cds_rel_rod'),\n",
       "  ('ENA Coding Sequence Release Synthetic', 'em_cds_rel_syn'),\n",
       "  ('ENA Coding Sequence Release Transgenic', 'em_cds_rel_tgn'),\n",
       "  ('ENA Coding Sequence Release Unclassified', 'em_cds_rel_unc'),\n",
       "  ('ENA Coding Sequence Release Viral', 'em_cds_rel_vrl'),\n",
       "  ('ENA Coding Sequence Release Vertebrate', 'em_cds_rel_vrt'),\n",
       "  ('ENA Coding Sequence Release Contigs Environment', 'em_cds_rel_con_env'),\n",
       "  ('ENA Coding Sequence Release Contigs Fungi', 'em_cds_rel_con_fun'),\n",
       "  ('ENA Coding Sequence Release Contigs Human', 'em_cds_rel_con_hum'),\n",
       "  ('ENA Coding Sequence Release Contigs Invertebrate', 'em_cds_rel_con_inv'),\n",
       "  ('ENA Coding Sequence Release Contigs Mammal', 'em_cds_rel_con_mam'),\n",
       "  ('ENA Coding Sequence Release Contigs Mouse', 'em_cds_rel_con_mus'),\n",
       "  ('ENA Coding Sequence Release Contigs Plant', 'em_cds_rel_con_pln'),\n",
       "  ('ENA Coding Sequence Release Contigs Prokaryote', 'em_cds_rel_con_pro'),\n",
       "  ('ENA Coding Sequence Release Contigs Rodent', 'em_cds_rel_con_rod'),\n",
       "  ('ENA Coding Sequence Release Contigs Vertebrate', 'em_cds_rel_con_vrt'),\n",
       "  ('ENA Coding Sequence Release EST Invertebrate', 'em_cds_rel_est_inv'),\n",
       "  ('ENA Coding Sequence Release EST Mammal', 'em_cds_rel_est_mam'),\n",
       "  ('ENA Coding Sequence Release EST Plant', 'em_cds_rel_est_pln'),\n",
       "  ('ENA Coding Sequence Release GSS Environment', 'em_cds_rel_gss_env'),\n",
       "  ('ENA Coding Sequence Release GSS Plant', 'em_cds_rel_gss_pln'),\n",
       "  ('ENA Coding Sequence Release GSS Prokaryote', 'em_cds_rel_gss_pro'),\n",
       "  ('ENA Coding Sequence Release HTC Environment', 'em_cds_rel_htc_env'),\n",
       "  ('ENA Coding Sequence Release HTC Fungi', 'em_cds_rel_htc_fun'),\n",
       "  ('ENA Coding Sequence Release HTC Human', 'em_cds_rel_htc_hum'),\n",
       "  ('ENA Coding Sequence Release HTC Invertebrate', 'em_cds_rel_htc_inv'),\n",
       "  ('ENA Coding Sequence Release HTC Mammal', 'em_cds_rel_htc_mam'),\n",
       "  ('ENA Coding Sequence Release HTC Mouse', 'em_cds_rel_htc_mus'),\n",
       "  ('ENA Coding Sequence Release HTC Plant', 'em_cds_rel_htc_pln'),\n",
       "  ('ENA Coding Sequence Release HTC Prokaryote', 'em_cds_rel_htc_pro'),\n",
       "  ('ENA Coding Sequence Release HTC Rodent', 'em_cds_rel_htc_rod'),\n",
       "  ('ENA Coding Sequence Release HTC Vertebrate', 'em_cds_rel_htc_vrt'),\n",
       "  ('ENA Coding Sequence Release HTG Environment', 'em_cds_rel_htg_env'),\n",
       "  ('ENA Coding Sequence Release HTG Fungi', 'em_cds_rel_htg_fun'),\n",
       "  ('ENA Coding Sequence Release HTG Human', 'em_cds_rel_htg_hum'),\n",
       "  ('ENA Coding Sequence Release HTG Invertebrate', 'em_cds_rel_htg_inv'),\n",
       "  ('ENA Coding Sequence Release HTG Mammal', 'em_cds_rel_htg_mam'),\n",
       "  ('ENA Coding Sequence Release HTG Mouse', 'em_cds_rel_htg_mus'),\n",
       "  ('ENA Coding Sequence Release HTG Phage', 'em_cds_rel_htg_phg'),\n",
       "  ('ENA Coding Sequence Release HTG Plant', 'em_cds_rel_htg_pln'),\n",
       "  ('ENA Coding Sequence Release HTG Prokaryote', 'em_cds_rel_htg_pro'),\n",
       "  ('ENA Coding Sequence Release HTG Rodent', 'em_cds_rel_htg_rod'),\n",
       "  ('ENA Coding Sequence Release HTG Viral', 'em_cds_rel_htg_vrl'),\n",
       "  ('ENA Coding Sequence Release HTG Vertebrate', 'em_cds_rel_htg_vrt'),\n",
       "  ('ENA Coding Sequence Release Patent Environment', 'em_cds_rel_pat_env'),\n",
       "  ('ENA Coding Sequence Release Patent Fungi', 'em_cds_rel_pat_fun'),\n",
       "  ('ENA Coding Sequence Release Patent Human', 'em_cds_rel_pat_hum'),\n",
       "  ('ENA Coding Sequence Release Patent Invertebrate', 'em_cds_rel_pat_inv'),\n",
       "  ('ENA Coding Sequence Release Patent Mammal', 'em_cds_rel_pat_mam'),\n",
       "  ('ENA Coding Sequence Release Patent Mouse', 'em_cds_rel_pat_mus'),\n",
       "  ('ENA Coding Sequence Release Patent Phage', 'em_cds_rel_pat_phg'),\n",
       "  ('ENA Coding Sequence Release Patent Plant', 'em_cds_rel_pat_pln'),\n",
       "  ('ENA Coding Sequence Release Patent Prokaryote', 'em_cds_rel_pat_pro'),\n",
       "  ('ENA Coding Sequence Release Patent Rodent', 'em_cds_rel_pat_rod'),\n",
       "  ('ENA Coding Sequence Release Patent Synthetic', 'em_cds_rel_pat_syn'),\n",
       "  ('ENA Coding Sequence Release Patent Unclassified', 'em_cds_rel_pat_unc'),\n",
       "  ('ENA Coding Sequence Release Patent Viral', 'em_cds_rel_pat_vrl'),\n",
       "  ('ENA Coding Sequence Release Patent Vertebrate', 'em_cds_rel_pat_vrt'),\n",
       "  ('ENA Coding Sequence Release Standard Environment', 'em_cds_rel_std_env'),\n",
       "  ('ENA Coding Sequence Release Standard Fungi', 'em_cds_rel_std_fun'),\n",
       "  ('ENA Coding Sequence Release Standard Human', 'em_cds_rel_std_hum'),\n",
       "  ('ENA Coding Sequence Release Standard Invertebrate', 'em_cds_rel_std_inv'),\n",
       "  ('ENA Coding Sequence Release Standard Mammal', 'em_cds_rel_std_mam'),\n",
       "  ('ENA Coding Sequence Release Standard Mouse', 'em_cds_rel_std_mus'),\n",
       "  ('ENA Coding Sequence Release Standard Phage', 'em_cds_rel_std_phg'),\n",
       "  ('ENA Coding Sequence Release Standard Plant', 'em_cds_rel_std_pln'),\n",
       "  ('ENA Coding Sequence Release Standard Prokaryote', 'em_cds_rel_std_pro'),\n",
       "  ('ENA Coding Sequence Release Standard Rodent', 'em_cds_rel_std_rod'),\n",
       "  ('ENA Coding Sequence Release Standard Synthetic', 'em_cds_rel_std_syn'),\n",
       "  ('ENA Coding Sequence Release Standard Transgenic', 'em_cds_rel_std_tgn'),\n",
       "  ('ENA Coding Sequence Release Standard Unclassified', 'em_cds_rel_std_unc'),\n",
       "  ('ENA Coding Sequence Release Standard Viral', 'em_cds_rel_std_vrl'),\n",
       "  ('ENA Coding Sequence Release Standard Vertebrate', 'em_cds_rel_std_vrt'),\n",
       "  ('ENA Coding Sequence Release STS Human', 'em_cds_rel_sts_hum'),\n",
       "  ('ENA Coding Sequence Release STS Mammal', 'em_cds_rel_sts_mam'),\n",
       "  ('ENA Coding Sequence Release STS Plant', 'em_cds_rel_sts_pln'),\n",
       "  ('ENA Coding Sequence Release STS Vertebrate', 'em_cds_rel_sts_vrt'),\n",
       "  ('ENA Coding Sequence Release TSA Fungi', 'em_cds_rel_tsa_fun'),\n",
       "  ('ENA Coding Sequence Release TSA Invertebrate', 'em_cds_rel_tsa_inv'),\n",
       "  ('ENA Coding Sequence Release TSA Mammal', 'em_cds_rel_tsa_mam'),\n",
       "  ('ENA Coding Sequence Release TSA Plant', 'em_cds_rel_tsa_pln'),\n",
       "  ('ENA Coding Sequence Release TSA Rodent', 'em_cds_rel_tsa_rod'),\n",
       "  ('ENA Coding Sequence Release TSA Vertebrate', 'em_cds_rel_tsa_vrt'),\n",
       "  ('ENA Coding Sequence Release WGS Environment', 'em_cds_rel_wgs_env'),\n",
       "  ('ENA Coding Sequence Release WGS Fungi', 'em_cds_rel_wgs_fun'),\n",
       "  ('ENA Coding Sequence Release WGS Invertebrate', 'em_cds_rel_wgs_inv'),\n",
       "  ('ENA Coding Sequence Release WGS Plant', 'em_cds_rel_wgs_pln'),\n",
       "  ('ENA Coding Sequence Release WGS Prokaryote', 'em_cds_rel_wgs_pro'),\n",
       "  ('ENA Coding Sequence Release WGS Viral', 'em_cds_rel_wgs_vrl'),\n",
       "  ('ENA Coding Sequence Release WGS Vertebrate', 'em_cds_rel_wgs_vrt'),\n",
       "  ('ENA Coding Sequence Updates Environment', 'em_cds_cum_env'),\n",
       "  ('ENA Coding Sequence Updates Fungi', 'em_cds_cum_fun'),\n",
       "  ('ENA Coding Sequence Updates Human', 'em_cds_cum_hum'),\n",
       "  ('ENA Coding Sequence Updates Invertebrate', 'em_cds_cum_inv'),\n",
       "  ('ENA Coding Sequence Updates Mammal', 'em_cds_cum_mam'),\n",
       "  ('ENA Coding Sequence Updates Mouse', 'em_cds_cum_mus'),\n",
       "  ('ENA Coding Sequence Updates Patent', 'em_cds_cum_pat'),\n",
       "  ('ENA Coding Sequence Updates Phage', 'em_cds_cum_phg'),\n",
       "  ('ENA Coding Sequence Updates Plant', 'em_cds_cum_pln'),\n",
       "  ('ENA Coding Sequence Updates Prokaryote', 'em_cds_cum_pro'),\n",
       "  ('ENA Coding Sequence Updates Rodent', 'em_cds_cum_rod'),\n",
       "  ('ENA Coding Sequence Updates Synthetic', 'em_cds_cum_syn'),\n",
       "  ('ENA Coding Sequence Updates Transgenic', 'em_cds_cum_tgn'),\n",
       "  ('ENA Coding Sequence Updates Unclassified', 'em_cds_cum_unc'),\n",
       "  ('ENA Coding Sequence Updates Viral', 'em_cds_cum_vrl'),\n",
       "  ('ENA Coding Sequence Updates Vertebrate', 'em_cds_cum_vrt'),\n",
       "  ('ENA Coding Sequence Updates Contigs', 'em_cds_cum_con'),\n",
       "  ('ENA Coding Sequence Updates Contigs Environment', 'em_cds_cum_con_env'),\n",
       "  ('ENA Coding Sequence Updates Contigs Fungi', 'em_cds_cum_con_fun'),\n",
       "  ('ENA Coding Sequence Updates Contigs Human', 'em_cds_cum_con_hum'),\n",
       "  ('ENA Coding Sequence Updates Contigs Invertebrate', 'em_cds_cum_con_inv'),\n",
       "  ('ENA Coding Sequence Updates Contigs Mammal', 'em_cds_cum_con_mam'),\n",
       "  ('ENA Coding Sequence Updates Contigs Mouse', 'em_cds_cum_con_mus'),\n",
       "  ('ENA Coding Sequence Updates Contigs Plant', 'em_cds_cum_con_pln'),\n",
       "  ('ENA Coding Sequence Updates Contigs Prokaryote', 'em_cds_cum_con_pro'),\n",
       "  ('ENA Coding Sequence Updates Contigs Rodent', 'em_cds_cum_con_rod'),\n",
       "  ('ENA Coding Sequence Updates Contigs Vertebrate', 'em_cds_cum_con_vrt'),\n",
       "  ('ENA Coding Sequence Updates EST Invertebrate', 'em_cds_cum_est_inv'),\n",
       "  ('ENA Coding Sequence Updates EST Mammal', 'em_cds_cum_est_mam'),\n",
       "  ('ENA Coding Sequence Updates EST Plant', 'em_cds_cum_est_pln'),\n",
       "  ('ENA Coding Sequence Updates GSS Environment', 'em_cds_cum_gss_env'),\n",
       "  ('ENA Coding Sequence Updates GSS Plant', 'em_cds_cum_gss_pln'),\n",
       "  ('ENA Coding Sequence Updates GSS Prokaryote', 'em_cds_cum_gss_pro'),\n",
       "  ('ENA Coding Sequence Updates HTC Environment', 'em_cds_cum_htc_env'),\n",
       "  ('ENA Coding Sequence Updates HTC Fungi', 'em_cds_cum_htc_fun'),\n",
       "  ('ENA Coding Sequence Updates HTC Human', 'em_cds_cum_htc_hum'),\n",
       "  ('ENA Coding Sequence Updates HTC Invertebrate', 'em_cds_cum_htc_inv'),\n",
       "  ('ENA Coding Sequence Updates HTC Mammal', 'em_cds_cum_htc_mam'),\n",
       "  ('ENA Coding Sequence Updates HTC Mouse', 'em_cds_cum_htc_mus'),\n",
       "  ('ENA Coding Sequence Updates HTC Plant', 'em_cds_cum_htc_pln'),\n",
       "  ('ENA Coding Sequence Updates HTC Prokaryote', 'em_cds_cum_htc_pro'),\n",
       "  ('ENA Coding Sequence Updates HTC Rodent', 'em_cds_cum_htc_rod'),\n",
       "  ('ENA Coding Sequence Updates HTC Vertebrate', 'em_cds_cum_htc_vrt'),\n",
       "  ('ENA Coding Sequence Updates HTG Environment', 'em_cds_cum_htg_env'),\n",
       "  ('ENA Coding Sequence Updates HTG Fungi', 'em_cds_cum_htg_fun'),\n",
       "  ('ENA Coding Sequence Updates HTG Human', 'em_cds_cum_htg_hum'),\n",
       "  ('ENA Coding Sequence Updates HTG Invertebrate', 'em_cds_cum_htg_inv'),\n",
       "  ('ENA Coding Sequence Updates HTG Mammal', 'em_cds_cum_htg_mam'),\n",
       "  ('ENA Coding Sequence Updates HTG Mouse', 'em_cds_cum_htg_mus'),\n",
       "  ('ENA Coding Sequence Updates HTG Phage', 'em_cds_cum_htg_phg'),\n",
       "  ('ENA Coding Sequence Updates HTG Plant', 'em_cds_cum_htg_pln'),\n",
       "  ('ENA Coding Sequence Updates HTG Prokaryote', 'em_cds_cum_htg_pro'),\n",
       "  ('ENA Coding Sequence Updates HTG Rodent', 'em_cds_cum_htg_rod'),\n",
       "  ('ENA Coding Sequence Updates HTG Viral', 'em_cds_cum_htg_vrl'),\n",
       "  ('ENA Coding Sequence Updates HTG Vertebrate', 'em_cds_cum_htg_vrt'),\n",
       "  ('ENA Coding Sequence Updates Patent Environment', 'em_cds_cum_pat_env'),\n",
       "  ('ENA Coding Sequence Updates Patent Fungi', 'em_cds_cum_pat_fun'),\n",
       "  ('ENA Coding Sequence Updates Patent Human', 'em_cds_cum_pat_hum'),\n",
       "  ('ENA Coding Sequence Updates Patent Invertebrate', 'em_cds_cum_pat_inv'),\n",
       "  ('ENA Coding Sequence Updates Patent Mammal', 'em_cds_cum_pat_mam'),\n",
       "  ('ENA Coding Sequence Updates Patent Mouse', 'em_cds_cum_pat_mus'),\n",
       "  ('ENA Coding Sequence Updates Patent Phage', 'em_cds_cum_pat_phg'),\n",
       "  ('ENA Coding Sequence Updates Patent Plant', 'em_cds_cum_pat_pln'),\n",
       "  ('ENA Coding Sequence Updates Patent Prokaryote', 'em_cds_cum_pat_pro'),\n",
       "  ('ENA Coding Sequence Updates Patent Rodent', 'em_cds_cum_pat_rod'),\n",
       "  ('ENA Coding Sequence Updates Patent Synthetic', 'em_cds_cum_pat_syn'),\n",
       "  ('ENA Coding Sequence Updates Patent Unclassified', 'em_cds_cum_pat_unc'),\n",
       "  ('ENA Coding Sequence Updates Patent Viral', 'em_cds_cum_pat_vrl'),\n",
       "  ('ENA Coding Sequence Updates Patent Vertebrate', 'em_cds_cum_pat_vrt'),\n",
       "  ('ENA Coding Sequence Updates Standard', 'em_cds_cum_std'),\n",
       "  ('ENA Coding Sequence Updates Standard Environment', 'em_cds_cum_std_env'),\n",
       "  ('ENA Coding Sequence Updates Standard Fungi', 'em_cds_cum_std_fun'),\n",
       "  ('ENA Coding Sequence Updates Standard Human', 'em_cds_cum_std_hum'),\n",
       "  ('ENA Coding Sequence Updates Standard Invertebrate', 'em_cds_cum_std_inv'),\n",
       "  ('ENA Coding Sequence Updates Standard Mammal', 'em_cds_cum_std_mam'),\n",
       "  ('ENA Coding Sequence Updates Standard Mouse', 'em_cds_cum_std_mus'),\n",
       "  ('ENA Coding Sequence Updates Standard Phage', 'em_cds_cum_std_phg'),\n",
       "  ('ENA Coding Sequence Updates Standard Plant', 'em_cds_cum_std_pln'),\n",
       "  ('ENA Coding Sequence Updates Standard Prokaryote', 'em_cds_cum_std_pro'),\n",
       "  ('ENA Coding Sequence Updates Standard Rodent', 'em_cds_cum_std_rod'),\n",
       "  ('ENA Coding Sequence Updates Standard Synthetic', 'em_cds_cum_std_syn'),\n",
       "  ('ENA Coding Sequence Updates Standard Transgenic', 'em_cds_cum_std_tgn'),\n",
       "  ('ENA Coding Sequence Updates Standard Unclassified', 'em_cds_cum_std_unc'),\n",
       "  ('ENA Coding Sequence Updates Standard Viral', 'em_cds_cum_std_vrl'),\n",
       "  ('ENA Coding Sequence Updates Standard Vertebrate', 'em_cds_cum_std_vrt'),\n",
       "  ('ENA Coding Sequence Updates STS Human', 'em_cds_cum_sts_hum'),\n",
       "  ('ENA Coding Sequence Updates STS Mammal', 'em_cds_cum_sts_mam'),\n",
       "  ('ENA Coding Sequence Updates STS Plant', 'em_cds_cum_sts_pln'),\n",
       "  ('ENA Coding Sequence Updates STS Vertebrate', 'em_cds_cum_sts_vrt'),\n",
       "  ('ENA Coding Sequence Updates TSA', 'em_cds_cum_tsa'),\n",
       "  ('ENA Coding Sequence Updates TSA Fungi', 'em_cds_cum_tsa_fun'),\n",
       "  ('ENA Coding Sequence Updates TSA Invertebrate', 'em_cds_cum_tsa_inv'),\n",
       "  ('ENA Coding Sequence Updates TSA Mammal', 'em_cds_cum_tsa_mam'),\n",
       "  ('ENA Coding Sequence Updates TSA Plant', 'em_cds_cum_tsa_pln'),\n",
       "  ('ENA Coding Sequence Updates TSA Rodent', 'em_cds_cum_tsa_rod'),\n",
       "  ('ENA Coding Sequence Updates TSA Vertebrate', 'em_cds_cum_tsa_vrt'),\n",
       "  ('ENA Coding Sequence Updates WGS', 'em_cds_cum_wgs'),\n",
       "  ('ENA Coding Sequence Updates WGS Environment', 'em_cds_cum_wgs_env'),\n",
       "  ('ENA Coding Sequence Updates WGS Fungi', 'em_cds_cum_wgs_fun'),\n",
       "  ('ENA Coding Sequence Updates WGS Invertebrate', 'em_cds_cum_wgs_inv'),\n",
       "  ('ENA Coding Sequence Updates WGS Plant', 'em_cds_cum_wgs_pln'),\n",
       "  ('ENA Coding Sequence Updates WGS Prokaryote', 'em_cds_cum_wgs_pro'),\n",
       "  ('ENA Coding Sequence Updates WGS Viral', 'em_cds_cum_wgs_vrl'),\n",
       "  ('ENA Coding Sequence Updates WGS Vertebrate', 'em_cds_cum_wgs_vrt'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs', 'em_ncr_rel_con'),\n",
       "  ('ENA Non-Coding Sequence Release EST', 'em_ncr_rel_est'),\n",
       "  ('ENA Non-Coding Sequence Release GSS', 'em_ncr_rel_gss'),\n",
       "  ('ENA Non-Coding Sequence Release HTC', 'em_ncr_rel_htc'),\n",
       "  ('ENA Non-Coding Sequence Release HTG', 'em_ncr_rel_htg'),\n",
       "  ('ENA Non-Coding Sequence Release Patent', 'em_ncr_rel_pat'),\n",
       "  ('ENA Non-Coding Sequence Release Standard', 'em_ncr_rel_std'),\n",
       "  ('ENA Non-Coding Sequence Release STS', 'em_ncr_rel_sts'),\n",
       "  ('ENA Non-Coding Sequence Release TSA', 'em_ncr_rel_tsa'),\n",
       "  ('ENA Non-Coding Sequence Release WGS', 'em_ncr_rel_wgs'),\n",
       "  ('ENA Non-Coding Sequence Release Environment', 'em_ncr_rel_env'),\n",
       "  ('ENA Non-Coding Sequence Release Fungi', 'em_ncr_rel_fun'),\n",
       "  ('ENA Non-Coding Sequence Release Human', 'em_ncr_rel_hum'),\n",
       "  ('ENA Non-Coding Sequence Release Invertebrate', 'em_ncr_rel_inv'),\n",
       "  ('ENA Non-Coding Sequence Release Mammal', 'em_ncr_rel_mam'),\n",
       "  ('ENA Non-Coding Sequence Release Mouse', 'em_ncr_rel_mus'),\n",
       "  ('ENA Non-Coding Sequence Release Phage', 'em_ncr_rel_phg'),\n",
       "  ('ENA Non-Coding Sequence Release Plant', 'em_ncr_rel_pln'),\n",
       "  ('ENA Non-Coding Sequence Release Prokaryote', 'em_ncr_rel_pro'),\n",
       "  ('ENA Non-Coding Sequence Release Rodent', 'em_ncr_rel_rod'),\n",
       "  ('ENA Non-Coding Sequence Release Synthetic', 'em_ncr_rel_syn'),\n",
       "  ('ENA Non-Coding Sequence Release Transgenic', 'em_ncr_rel_tgn'),\n",
       "  ('ENA Non-Coding Sequence Release Unclassified', 'em_ncr_rel_unc'),\n",
       "  ('ENA Non-Coding Sequence Release Viral', 'em_ncr_rel_vrl'),\n",
       "  ('ENA Non-Coding Sequence Release Vertebrate', 'em_ncr_rel_vrt'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Environment',\n",
       "   'em_ncr_rel_con_env'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Fungi', 'em_ncr_rel_con_fun'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Invertebrate',\n",
       "   'em_ncr_rel_con_inv'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Mammal', 'em_ncr_rel_con_mam'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Plant', 'em_ncr_rel_con_pln'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Prokaryote', 'em_ncr_rel_con_pro'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Rodent', 'em_ncr_rel_con_rod'),\n",
       "  ('ENA Non-Coding Sequence Release Contigs Vertebrate', 'em_ncr_rel_con_vrt'),\n",
       "  ('ENA Non-Coding Sequence Release EST Mammal', 'em_ncr_rel_est_mam'),\n",
       "  ('ENA Non-Coding Sequence Release EST Rodent', 'em_ncr_rel_est_rod'),\n",
       "  ('ENA Non-Coding Sequence Release EST Vertebrate', 'em_ncr_rel_est_vrt'),\n",
       "  ('ENA Non-Coding Sequence Release GSS Fungi', 'em_ncr_rel_gss_fun'),\n",
       "  ('ENA Non-Coding Sequence Release GSS Invertebrate', 'em_ncr_rel_gss_inv'),\n",
       "  ('ENA Non-Coding Sequence Release GSS Human', 'em_ncr_rel_htc_hum'),\n",
       "  ('ENA Non-Coding Sequence Release HTC Invertebrate', 'em_ncr_rel_htc_inv'),\n",
       "  ('ENA Non-Coding Sequence Release HTC Plant', 'em_ncr_rel_htc_pln'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Environment', 'em_ncr_rel_htg_env'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Invertebrate', 'em_ncr_rel_htg_inv'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Mammal', 'em_ncr_rel_htg_mam'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Mouse', 'em_ncr_rel_htg_mus'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Phage', 'em_ncr_rel_htg_phg'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Plant', 'em_ncr_rel_htg_pln'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Prokaryote', 'em_ncr_rel_htg_pro'),\n",
       "  ('ENA Non-Coding Sequence Release HTG Viral', 'em_ncr_rel_htg_vrl'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Fungi', 'em_ncr_rel_pat_fun'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Human', 'em_ncr_rel_pat_hum'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Invertebrate',\n",
       "   'em_ncr_rel_pat_inv'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Plant', 'em_ncr_rel_pat_pln'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Prokaryote', 'em_ncr_rel_pat_pro'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Synthetic', 'em_ncr_rel_pat_syn'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Unclassified',\n",
       "   'em_ncr_rel_pat_unc'),\n",
       "  ('ENA Non-Coding Sequence Release Patent Viral', 'em_ncr_rel_pat_vrl'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Environment',\n",
       "   'em_ncr_rel_std_env'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Fungi', 'em_ncr_rel_std_fun'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Human', 'em_ncr_rel_std_hum'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Invertebrate',\n",
       "   'em_ncr_rel_std_inv'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Mammal', 'em_ncr_rel_std_mam'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Mouse', 'em_ncr_rel_std_mus'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Phage', 'em_ncr_rel_std_phg'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Plant', 'em_ncr_rel_std_pln'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Prokaryote',\n",
       "   'em_ncr_rel_std_pro'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Rodent', 'em_ncr_rel_std_rod'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Synthetic', 'em_ncr_rel_std_syn'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Transgenic',\n",
       "   'em_ncr_rel_std_tgn'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Unclassified',\n",
       "   'em_ncr_rel_std_unc'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Viral', 'em_ncr_rel_std_vrl'),\n",
       "  ('ENA Non-Coding Sequence Release Standard Vertebrate',\n",
       "   'em_ncr_rel_std_vrt'),\n",
       "  ('ENA Non-Coding Sequence Release STS Fungi', 'em_ncr_rel_sts_fun'),\n",
       "  ('ENA Non-Coding Sequence Release STS Human', 'em_ncr_rel_sts_hum'),\n",
       "  ('ENA Non-Coding Sequence Release STS Vertebrate', 'em_ncr_rel_sts_vrt'),\n",
       "  ('ENA Non-Coding Sequence Release TSA Human', 'em_ncr_rel_tsa_hum'),\n",
       "  ('ENA Non-Coding Sequence Release TSA Invertebrate', 'em_ncr_rel_tsa_inv'),\n",
       "  ('ENA Non-Coding Sequence Release TSA Mammal', 'em_ncr_rel_tsa_mam'),\n",
       "  ('ENA Non-Coding Sequence Release TSA Mouse', 'em_ncr_rel_tsa_mus'),\n",
       "  ('ENA Non-Coding Sequence Release TSA Plant', 'em_ncr_rel_tsa_pln'),\n",
       "  ('ENA Non-Coding Sequence Release WGS Environment', 'em_ncr_rel_wgs_env'),\n",
       "  ('ENA Non-Coding Sequence Release WGS Fungi', 'em_ncr_rel_wgs_fun'),\n",
       "  ('ENA Non-Coding Sequence Release WGS Invertebrate', 'em_ncr_rel_wgs_inv'),\n",
       "  ('ENA Non-Coding Sequence Release WGS Plant', 'em_ncr_rel_wgs_pln'),\n",
       "  ('ENA Non-Coding Sequence Release WGS Prokaryote', 'em_ncr_rel_wgs_pro'),\n",
       "  ('ENA Non-Coding Sequence Release WGS Viral', 'em_ncr_rel_wgs_vrl'),\n",
       "  ('ENA Non-Coding Sequence Updates Environment', 'em_ncr_cum_env'),\n",
       "  ('ENA Non-Coding Sequence Updates Fungi', 'em_ncr_cum_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates Human', 'em_ncr_cum_hum'),\n",
       "  ('ENA Non-Coding Sequence Updates Invertebrate', 'em_ncr_cum_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates Mammal', 'em_ncr_cum_mam'),\n",
       "  ('ENA Non-Coding Sequence Updates Mouse', 'em_ncr_cum_mus'),\n",
       "  ('ENA Non-Coding Sequence Updates Phage', 'em_ncr_cum_phg'),\n",
       "  ('ENA Non-Coding Sequence Updates Plant', 'em_ncr_cum_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates Prokaryote', 'em_ncr_cum_pro'),\n",
       "  ('ENA Non-Coding Sequence Updates Rodent', 'em_ncr_cum_rod'),\n",
       "  ('ENA Non-Coding Sequence Updates Synthetic', 'em_ncr_cum_syn'),\n",
       "  ('ENA Non-Coding Sequence Updates Unclassified', 'em_ncr_cum_unc'),\n",
       "  ('ENA Non-Coding Sequence Updates Vertebrate', 'em_ncr_cum_vrt'),\n",
       "  ('ENA Non-Coding Sequence Updates Viral', 'em_ncr_cum_vrl'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs', 'em_ncr_cum_con'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Environment',\n",
       "   'em_ncr_cum_con_env'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Fungi', 'em_ncr_cum_con_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Invertebrate',\n",
       "   'em_ncr_cum_con_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Mammal', 'em_ncr_cum_con_mam'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Plant', 'em_ncr_cum_con_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Prokaryote', 'em_ncr_cum_con_pro'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Rodent', 'em_ncr_cum_con_rod'),\n",
       "  ('ENA Non-Coding Sequence Updates Contigs Vertebrate', 'em_ncr_cum_con_vrt'),\n",
       "  ('ENA Non-Coding Sequence Updates EST Mammal', 'em_ncr_cum_est_mam'),\n",
       "  ('ENA Non-Coding Sequence Updates EST Rodent', 'em_ncr_cum_est_rod'),\n",
       "  ('ENA Non-Coding Sequence Updates EST Vertebrate', 'em_ncr_cum_est_vrt'),\n",
       "  ('ENA Non-Coding Sequence Updates GSS Fungi', 'em_ncr_cum_gss_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates GSS Invertebrate', 'em_ncr_cum_gss_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates GSS Human', 'em_ncr_cum_htc_hum'),\n",
       "  ('ENA Non-Coding Sequence Updates HTC Invertebrate', 'em_ncr_cum_htc_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates HTC Plant', 'em_ncr_cum_htc_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Environment', 'em_ncr_cum_htg_env'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Invertebrate', 'em_ncr_cum_htg_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Mammal', 'em_ncr_cum_htg_mam'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Mouse', 'em_ncr_cum_htg_mus'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Phage', 'em_ncr_cum_htg_phg'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Plant', 'em_ncr_cum_htg_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Prokaryote', 'em_ncr_cum_htg_pro'),\n",
       "  ('ENA Non-Coding Sequence Updates HTG Viral', 'em_ncr_cum_htg_vrl'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Fungi', 'em_ncr_cum_pat_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Human', 'em_ncr_cum_pat_hum'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Invertebrate',\n",
       "   'em_ncr_cum_pat_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Plant', 'em_ncr_cum_pat_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Prokaryote', 'em_ncr_cum_pat_pro'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Synthetic', 'em_ncr_cum_pat_syn'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Unclassified',\n",
       "   'em_ncr_cum_pat_unc'),\n",
       "  ('ENA Non-Coding Sequence Updates Patent Viral', 'em_ncr_cum_pat_vrl'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard', 'em_ncr_cum_std'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Environment',\n",
       "   'em_ncr_cum_std_env'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Fungi', 'em_ncr_cum_std_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Human', 'em_ncr_cum_std_hum'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Invertebrate',\n",
       "   'em_ncr_cum_std_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Mammal', 'em_ncr_cum_std_mam'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Mouse', 'em_ncr_cum_std_mus'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Phage', 'em_ncr_cum_std_phg'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Plant', 'em_ncr_cum_std_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Prokaryote',\n",
       "   'em_ncr_cum_std_pro'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Rodent', 'em_ncr_cum_std_rod'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Synthetic', 'em_ncr_cum_std_syn'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Transgenic',\n",
       "   'em_ncr_cum_std_tgn'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Unclassified',\n",
       "   'em_ncr_cum_std_unc'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Viral', 'em_ncr_cum_std_vrl'),\n",
       "  ('ENA Non-Coding Sequence Updates Standard Vertebrate',\n",
       "   'em_ncr_cum_std_vrt'),\n",
       "  ('ENA Non-Coding Sequence Updates STS Fungi', 'em_ncr_cum_sts_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates STS Human', 'em_ncr_cum_sts_hum'),\n",
       "  ('ENA Non-Coding Sequence Updates STS Vertebrate', 'em_ncr_cum_sts_vrt'),\n",
       "  ('ENA Non-Coding Sequence Updates TSA', 'em_ncr_cum_tsa'),\n",
       "  ('ENA Non-Coding Sequence Updates TSA Human', 'em_ncr_cum_tsa_hum'),\n",
       "  ('ENA Non-Coding Sequence Updates TSA Invertebrate', 'em_ncr_cum_tsa_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates TSA Mammal', 'em_ncr_cum_tsa_mam'),\n",
       "  ('ENA Non-Coding Sequence Updates TSA Mouse', 'em_ncr_cum_tsa_mus'),\n",
       "  ('ENA Non-Coding Sequence Updates TSA Plant', 'em_ncr_cum_tsa_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS', 'em_ncr_cum_wgs'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS Environment', 'em_ncr_cum_wgs_env'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS Fungi', 'em_ncr_cum_wgs_fun'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS Invertebrate', 'em_ncr_cum_wgs_inv'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS Plant', 'em_ncr_cum_wgs_pln'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS Prokaryote', 'em_ncr_cum_wgs_pro'),\n",
       "  ('ENA Non-Coding Sequence Updates WGS Viral', 'em_ncr_cum_wgs_vrl'),\n",
       "  ('Ampittia dioscorides ', '509334-g'),\n",
       "  ('Anopheles gambiae str. PEST ', '180454-g'),\n",
       "  ('Arabidopsis thaliana ', '3702-g'),\n",
       "  ('Aspergillus fumigatus Af293 ', '330879-g'),\n",
       "  ('Aspergillus nidulans FGSC A4 ', '227321-g'),\n",
       "  ('Aspergillus niger ', '5061-g'),\n",
       "  ('Aspidytes niobe ', '183865-g'),\n",
       "  ('Babesia microti strain RI ', '1133968-g'),\n",
       "  ('Bos taurus ', '9913-g'),\n",
       "  ('Brachypodium distachyon ', '15368-g'),\n",
       "  ('Caenorhabditis briggsae ', '6238-g'),\n",
       "  ('Caenorhabditis elegans ', '6239-g'),\n",
       "  ('Callithrix jacchus ', '9483-g'),\n",
       "  ('Candida albicans WO-1 ', '294748-g'),\n",
       "  ('Candida dubliniensis CD36 ', '573826-g'),\n",
       "  ('Candida glabrata ', '5478-g'),\n",
       "  ('Canis lupus familiaris ', '9615-g'),\n",
       "  ('Cannabis sativa ', '3483-g'),\n",
       "  ('Cherax cainii ', '223846-g'),\n",
       "  ('Cherax cairnsensis ', '220736-g'),\n",
       "  ('Cherax crassimanus ', '191330-g'),\n",
       "  ('Cherax destructor ', '6723-g'),\n",
       "  ('Cherax dispar ', '99751-g'),\n",
       "  ('Cherax preissii ', '167887-g'),\n",
       "  ('Cherax quadricarinatus ', '27406-g'),\n",
       "  ('Cherax quinquecarinatus ', '99753-g'),\n",
       "  ('Cherax robustus ', '220734-g'),\n",
       "  ('Cherax sp. HG-2014a ', '1447911-g'),\n",
       "  ('Cherax sp. HG-2014b ', '1447910-g'),\n",
       "  ('Coccinella septempunctata ', '41139-g'),\n",
       "  ('Cotton leaf curl Gezira alphasatellite ', '679536-g'),\n",
       "  ('Cryptococcus neoformans var. neoformans B-3501A ', '283643-g'),\n",
       "  ('Cryptococcus neoformans var. neoformans JEC21 ', '214684-g'),\n",
       "  ('Cryptosporidium parvum ', '5807-g'),\n",
       "  ('Cryptosporidium parvum Iowa II ', '353152-g'),\n",
       "  ('Cucumis melo ', '3656-g'),\n",
       "  ('Cyanidioschyzon merolae ', '45157-g'),\n",
       "  ('Cyanidioschyzon merolae strain 10D ', '280699-g'),\n",
       "  ('Danio rerio ', '7955-g'),\n",
       "  ('Debaryomyces hansenii CBS767 ', '284592-g'),\n",
       "  ('Dictyostelium discoideum AX4 ', '352472-g'),\n",
       "  ('Drosophila melanogaster ', '7227-g'),\n",
       "  ('Drosophila pseudoobscura pseudoobscura ', '46245-g'),\n",
       "  ('Drosophila simulans ', '7240-g'),\n",
       "  ('Drosophila virilis ', '7244-g'),\n",
       "  ('Drosophila yakuba ', '7245-g'),\n",
       "  ('Ectocarpus siliculosus ', '2880-g'),\n",
       "  ('Eimeria tenella ', '5802-g'),\n",
       "  ('Encephalitozoon cuniculi GB-M1 ', '284813-g'),\n",
       "  ('Encephalitozoon intestinalis ATCC 50506 ', '876142-g'),\n",
       "  ('Engaeus cunicularius ', '99757-g'),\n",
       "  ('Engaeus lyelli ', '219696-g'),\n",
       "  ('Equus caballus ', '9796-g'),\n",
       "  ('Eremothecium cymbalariae DBVPG#7215 ', '931890-g'),\n",
       "  ('Eremothecium gossypii ATCC 10895 ', '284811-g'),\n",
       "  ('Euastacus yarraensis ', '72426-g'),\n",
       "  ('Fusarium graminearum ', '5518-g'),\n",
       "  ('Fusarium graminearum PH-1 ', '229533-g'),\n",
       "  ('Fusarium oxysporum f. sp. lycopersici 4287 ', '426428-g'),\n",
       "  ('Fusarium verticillioides 7600 ', '334819-g'),\n",
       "  ('Gadus morhua ', '8049-g'),\n",
       "  ('Gallus gallus ', '9031-g'),\n",
       "  ('Glycine max ', '3847-g'),\n",
       "  ('Gorilla gorilla gorilla ', '9595-g'),\n",
       "  ('Hemiselmis andersenii ', '464988-g'),\n",
       "  ('Hydropotes inermis argyropus ', '344906-g'),\n",
       "  ('Kazachstania africana CBS 2517 ', '1071382-g'),\n",
       "  ('Kazachstania naganishii CBS 8797 ', '1071383-g'),\n",
       "  ('Kluyveromyces lactis ', '28985-g'),\n",
       "  ('Kluyveromyces marxianus ', '4911-g'),\n",
       "  ('Kluyveromyces marxianus DMKU3-1042 ', '1003335-g'),\n",
       "  ('Komagataella pastoris CBS 7435 ', '981350-g'),\n",
       "  ('Komagataella pastoris GS115 ', '644223-g'),\n",
       "  ('Lachancea kluyveri NRRL Y-12651 ', '226302-g'),\n",
       "  ('Lachancea thermotolerans CBS 6340 ', '559295-g'),\n",
       "  ('Leishmania braziliensis MHOM/BR/75/M2904 ', '420245-g'),\n",
       "  ('Leishmania donovani ', '5661-g'),\n",
       "  ('Leishmania infantum JPCM5 ', '435258-g'),\n",
       "  ('Leishmania major strain Friedlin ', '347515-g'),\n",
       "  ('Leishmania mexicana MHOM/GT/2001/U1103 ', '929439-g'),\n",
       "  ('Lutraria rhynchaena ', '1380851-g'),\n",
       "  ('Macaca fascicularis ', '9541-g'),\n",
       "  ('Macaca mulatta ', '9544-g'),\n",
       "  ('Maccullochella peelii ', '135761-g'),\n",
       "  ('Macquaria australasica ', '135765-g'),\n",
       "  ('Magnaporthe oryzae 70-15 ', '242507-g'),\n",
       "  ('Malus domestica ', '3750-g'),\n",
       "  ('Medicago truncatula ', '3880-g'),\n",
       "  ('Micromonas sp. RCC299 ', '296587-g'),\n",
       "  ('Monodelphis domestica ', '13616-g'),\n",
       "  ('Mus musculus ', '10090-g'),\n",
       "  ('Myceliophthora thermophila ATCC 42464 ', '573729-g'),\n",
       "  ('Mythimna separata ', '271217-g'),\n",
       "  ('Nasonia vitripennis ', '7425-g'),\n",
       "  ('Naumovozyma castellii CBS 4309 ', '1064592-g'),\n",
       "  ('Naumovozyma dairenensis CBS 421 ', '1071378-g'),\n",
       "  ('Neospora caninum Liverpool ', '572307-g'),\n",
       "  ('Oncomelania hupensis hupensis ', '111462-g'),\n",
       "  ('Ornithorhynchus anatinus ', '9258-g'),\n",
       "  ('Oryctolagus cuniculus ', '9986-g'),\n",
       "  ('Oryza sativa Indica Group ', '39946-g'),\n",
       "  ('Oryza sativa Japonica Group ', '39947-g'),\n",
       "  ('Oryzias latipes ', '8090-g'),\n",
       "  ('Ostreococcus lucimarinus CCE9901 ', '436017-g'),\n",
       "  ('Ostreococcus tauri ', '70448-g'),\n",
       "  ('Ovis aries ', '9940-g'),\n",
       "  ('Pan troglodytes ', '9598-g'),\n",
       "  ('Pandarus rhincodonicus ', '1473543-g'),\n",
       "  ('Paramecium tetraurelia ', '5888-g'),\n",
       "  ('Pastinachus atrus ', '1053764-g'),\n",
       "  ('Penicillium rubens Wisconsin 54-1255 ', '500485-g'),\n",
       "  ('Phaeodactylum tricornutum CCAP 1055/1 ', '556484-g'),\n",
       "  ('Phasianus colchicus ', '9054-g'),\n",
       "  ('Plasmodium chabaudi chabaudi ', '31271-g'),\n",
       "  ('Plasmodium falciparum 3D7 ', '36329-g'),\n",
       "  ('Plasmodium knowlesi strain H ', '5851-g'),\n",
       "  ('Plasmodium vivax ', '5855-g'),\n",
       "  ('Pongo abelii ', '9601-g'),\n",
       "  ('Protobothrops xiangchengensis ', '313189-g'),\n",
       "  ('Rattus norvegicus ', '10116-g'),\n",
       "  ('Saccharomyces cerevisiae S288c ', '559292-g'),\n",
       "  ('Scheffersomyces stipitis CBS 6054 ', '322104-g'),\n",
       "  ('Schistosoma mansoni ', '6183-g'),\n",
       "  ('Schizosaccharomyces pombe ', '4896-g'),\n",
       "  ('Secale cereale ', '4550-g'),\n",
       "  ('Serinus canaria ', '9135-g'),\n",
       "  ('Siniperca chuatsi x Siniperca scherzeri ', '1443077-g'),\n",
       "  ('Siniperca scherzeri x Siniperca chuatsi ', '1443078-g'),\n",
       "  ('Solanum lycopersicum ', '4081-g'),\n",
       "  ('Solanum pennellii ', '28526-g'),\n",
       "  ('Sorghum bicolor ', '4558-g'),\n",
       "  ('Spirometra erinaceieuropaei ', '99802-g'),\n",
       "  ('Sus scrofa ', '9823-g'),\n",
       "  ('Taeniopygia guttata ', '59729-g'),\n",
       "  ('Tetrapisispora blattae CBS 6284 ', '1071380-g'),\n",
       "  ('Tetrapisispora phaffii CBS 4417 ', '1071381-g'),\n",
       "  ('Thalassiosira pseudonana CCMP1335 ', '296543-g'),\n",
       "  ('Theileria annulata ', '5874-g'),\n",
       "  ('Theileria orientalis strain Shintoku ', '869250-g'),\n",
       "  ('Thielavia terrestris NRRL 8126 ', '578455-g'),\n",
       "  ('Torulaspora delbrueckii ', '4950-g'),\n",
       "  ('Toxoplasma gondii RH ', '383379-g'),\n",
       "  ('Tribolium castaneum ', '7070-g'),\n",
       "  ('Triticum aestivum ', '4565-g'),\n",
       "  ('Triticum monococcum subsp. aegilopoides ', '52163-g'),\n",
       "  ('Trypanosoma brucei brucei TREU927 ', '185431-g'),\n",
       "  ('Trypanosoma brucei gambiense DAL972 ', '679716-g'),\n",
       "  ('Vitis vinifera ', '29760-g'),\n",
       "  ('Yarrowia lipolytica CLIB122 ', '284591-g'),\n",
       "  ('Yarrowia lipolytica WSH-Z06 ', '1437815-g'),\n",
       "  ('Zea mays ', '4577-g'),\n",
       "  ('Zygosaccharomyces rouxii ', '4956-g'),\n",
       "  ('Acidianus hospitalis W1 ', '933801-g'),\n",
       "  ('Acidilobus saccharovorans 345-15 ', '666510-g'),\n",
       "  ('Aciduliprofundum boonei T469 ', '439481-g'),\n",
       "  ('Aciduliprofundum sp. MAR08-339 ', '673860-g'),\n",
       "  ('Aeropyrum camini SY1 = JCM 12091 ', '1198449-g'),\n",
       "  ('Aeropyrum pernix K1 ', '272557-g'),\n",
       "  ('Archaeoglobus fulgidus DSM 4304 ', '224325-g'),\n",
       "  ('Archaeoglobus fulgidus DSM 8774 ', '1344584-g'),\n",
       "  ('Archaeoglobus profundus DSM 5631 ', '572546-g'),\n",
       "  ('Archaeoglobus sulfaticallidus PM70-1 ', '387631-g'),\n",
       "  ('Archaeoglobus veneficus SNP6 ', '693661-g'),\n",
       "  ('Caldisphaera lagunensis DSM 15908 ', '1056495-g'),\n",
       "  ('Caldivirga maquilingensis IC-167 ', '397948-g'),\n",
       "  ('Candidatus Caldiarchaeum subterraneum ', '311458-g'),\n",
       "  ('Candidatus Korarchaeum cryptofilum OPF8 ', '374847-g'),\n",
       "  ('Candidatus Methanomassiliicoccus intestinalis Issoire-Mx1 ', '1295009-g'),\n",
       "  ('Candidatus Methanomethylophilus alvus Mx1201 ', '1236689-g'),\n",
       "  ('Candidatus Methanoplasma termitum ', '1577791-g'),\n",
       "  ('Candidatus Nitrosopumilus koreensis AR1 ', '1229908-g'),\n",
       "  ('Candidatus Nitrosopumilus sp. AR2 ', '1229909-g'),\n",
       "  ('Candidatus Nitrosopumilus sp. D3C ', '1582439-g'),\n",
       "  ('Candidatus Nitrososphaera evergladensis SR1 ', '1459636-g'),\n",
       "  ('Candidatus Nitrososphaera gargensis Ga9.2 ', '1237085-g'),\n",
       "  ('Cenarchaeum symbiosum A ', '414004-g'),\n",
       "  ('Cotton leaf curl Gezira alphasatellite ', '679536-g'),\n",
       "  ('Desulfurococcus fermentans DSM 16532 ', '768672-g'),\n",
       "  ('Desulfurococcus kamchatkensis 1221n ', '490899-g'),\n",
       "  ('Desulfurococcus mucosus DSM 2162 ', '765177-g'),\n",
       "  ('Ferroglobus placidus DSM 10642 ', '589924-g'),\n",
       "  ('Ferroplasma acidarmanus fer1 ', '333146-g'),\n",
       "  ('Fervidicoccus fontis Kam940 ', '1163730-g'),\n",
       "  ('Geoglobus acetivorans ', '565033-g'),\n",
       "  ('Halalkalicoccus jeotgali B3 ', '795797-g'),\n",
       "  ('Haloarcula hispanica ATCC 33960 ', '634497-g'),\n",
       "  ('Haloarcula marismortui ATCC 43049 ', '272569-g'),\n",
       "  ('Haloarcula sp. CBA1115 ', '1592728-g'),\n",
       "  ('Halobacterium salinarum R1 ', '478009-g'),\n",
       "  ('Halobacterium sp. DL1 ', '751944-g'),\n",
       "  ('Halobacterium sp. NRC-1 ', '64091-g'),\n",
       "  ('Haloferax mediterranei ATCC 33500 ', '523841-g'),\n",
       "  ('Haloferax volcanii DS2 ', '309800-g'),\n",
       "  ('Halogeometricum borinquense DSM 11551 ', '469382-g'),\n",
       "  ('Halomicrobium mukohataei DSM 12286 ', '485914-g'),\n",
       "  ('Halopiger xanaduensis SH-6 ', '797210-g'),\n",
       "  ('Haloquadratum walsbyi C23 ', '768065-g'),\n",
       "  ('Haloquadratum walsbyi DSM 16790 ', '362976-g'),\n",
       "  ('Halorhabdus tiamatea SARL4B ', '1033806-g'),\n",
       "  ('Halorhabdus utahensis DSM 12940 ', '519442-g'),\n",
       "  ('Halorubrum lacusprofundi ATCC 49239 ', '416348-g'),\n",
       "  ('Halostagnicola larsenii XH-48 ', '797299-g'),\n",
       "  ('Haloterrigena turkmenica DSM 5511 ', '543526-g'),\n",
       "  ('Halovivax ruber XH-70 ', '797302-g'),\n",
       "  ('Hyperthermus butylicus DSM 5456 ', '415426-g'),\n",
       "  ('Ignicoccus hospitalis KIN4/I ', '453591-g'),\n",
       "  ('Ignisphaera aggregans DSM 17230 ', '583356-g'),\n",
       "  ('Metallosphaera cuprina Ar-4 ', '1006006-g'),\n",
       "  ('Metallosphaera sedula DSM 5348 ', '399549-g'),\n",
       "  ('Methanobacterium formicicum ', '2162-g'),\n",
       "  ('Methanobacterium lacus ', '877455-g'),\n",
       "  ('Methanobacterium paludis ', '868131-g'),\n",
       "  ('Methanobacterium sp. MB1 ', '1379702-g'),\n",
       "  ('Methanobrevibacter ruminantium M1 ', '634498-g'),\n",
       "  ('Methanobrevibacter smithii ATCC 35061 ', '420247-g'),\n",
       "  ('Methanobrevibacter sp. AbM4 ', '224719-g'),\n",
       "  ('Methanocaldococcus fervens AG86 ', '573064-g'),\n",
       "  ('Methanocaldococcus infernus ME ', '573063-g'),\n",
       "  ('Methanocaldococcus jannaschii DSM 2661 ', '243232-g'),\n",
       "  ('Methanocaldococcus sp. FS406-22 ', '644281-g'),\n",
       "  ('Methanocaldococcus sp. JH146 ', '1301915-g'),\n",
       "  ('Methanocaldococcus vulcanius M7 ', '579137-g'),\n",
       "  ('Methanocella arvoryzae MRE50 ', '351160-g'),\n",
       "  ('Methanocella conradii HZ254 ', '1041930-g'),\n",
       "  ('Methanocella paludicola SANAE ', '304371-g'),\n",
       "  ('Methanococcoides burtonii DSM 6242 ', '259564-g'),\n",
       "  ('Methanococcoides methylutens MM1 ', '1434104-g'),\n",
       "  ('Methanococcus aeolicus Nankai-3 ', '419665-g'),\n",
       "  ('Methanococcus maripaludis ', '39152-g'),\n",
       "  ('Methanococcus maripaludis C5 ', '402880-g'),\n",
       "  ('Methanococcus maripaludis C6 ', '444158-g'),\n",
       "  ('Methanococcus maripaludis C7 ', '426368-g'),\n",
       "  ('Methanococcus maripaludis X1 ', '1053692-g'),\n",
       "  ('Methanococcus vannielii SB ', '406327-g'),\n",
       "  ('Methanococcus voltae A3 ', '456320-g'),\n",
       "  ('Methanocorpusculum labreanum Z ', '410358-g'),\n",
       "  ('Methanoculleus bourgensis MS2 ', '1201294-g'),\n",
       "  ('Methanoculleus marisnigri JR1 ', '368407-g'),\n",
       "  ('Methanohalobium evestigatum Z-7303 ', '644295-g'),\n",
       "  ('Methanohalophilus mahii DSM 5219 ', '547558-g'),\n",
       "  ('Methanolacinia petrolearia DSM 11571 ', '679926-g'),\n",
       "  ('Methanolobus psychrophilus R15 ', '1094980-g'),\n",
       "  ('Methanomethylovorans hollandica DSM 15978 ', '867904-g'),\n",
       "  ('Methanopyrus kandleri AV19 ', '190192-g'),\n",
       "  ('Methanoregula boonei 6A8 ', '456442-g'),\n",
       "  ('Methanoregula formicica SMSP ', '593750-g'),\n",
       "  ('Methanosaeta concilii GP6 ', '990316-g'),\n",
       "  ('Methanosaeta harundinacea 6Ac ', '1110509-g'),\n",
       "  ('Methanosaeta thermophila PT ', '349307-g'),\n",
       "  ('Methanosalsum zhilinae DSM 4017 ', '679901-g'),\n",
       "  ('Methanosarcina acetivorans C2A ', '188937-g'),\n",
       "  ('Methanosarcina barkeri 227 ', '1434106-g'),\n",
       "  ('Methanosarcina barkeri 3 ', '1434107-g'),\n",
       "  ('Methanosarcina barkeri MS ', '1434108-g'),\n",
       "  ('Methanosarcina barkeri str. Fusaro ', '269797-g'),\n",
       "  ('Methanosarcina barkeri str. Wiesmoor ', '1434109-g'),\n",
       "  ('Methanosarcina horonobensis HB-1 = JCM 15518 ', '1434110-g'),\n",
       "  ('Methanosarcina lacustris Z-7289 ', '1434111-g'),\n",
       "  ('Methanosarcina mazei C16 ', '1434113-g'),\n",
       "  ('Methanosarcina mazei Go1 ', '192952-g'),\n",
       "  ('Methanosarcina mazei LYC ', '1434114-g'),\n",
       "  ('Methanosarcina mazei S-6 ', '213585-g'),\n",
       "  ('Methanosarcina mazei SarPi ', '1434115-g'),\n",
       "  ('Methanosarcina mazei Tuc01 ', '1236903-g'),\n",
       "  ('Methanosarcina mazei WWM610 ', '1434117-g'),\n",
       "  ('Methanosarcina siciliae C2J ', '1434118-g'),\n",
       "  ('Methanosarcina siciliae HI350 ', '1434119-g'),\n",
       "  ('Methanosarcina siciliae T4/M ', '1434120-g'),\n",
       "  ('Methanosarcina sp. Kolksee ', '1434099-g'),\n",
       "  ('Methanosarcina sp. MTP4 ', '1434100-g'),\n",
       "  ('Methanosarcina sp. WH1 ', '1434102-g'),\n",
       "  ('Methanosarcina sp. WWM596 ', '1434103-g'),\n",
       "  ('Methanosarcina thermophila CHTI-55 ', '1434121-g'),\n",
       "  ('Methanosarcina thermophila TM-1 ', '523844-g'),\n",
       "  ('Methanosarcina vacuolata Z-761 ', '1434123-g'),\n",
       "  ('Methanosphaera stadtmanae DSM 3091 ', '339860-g'),\n",
       "  ('Methanosphaerula palustris E1-9c ', '521011-g'),\n",
       "  ('Methanospirillum hungatei JF-1 ', '323259-g'),\n",
       "  ('Methanothermobacter marburgensis str. Marburg ', '79929-g'),\n",
       "  ('Methanothermobacter thermautotrophicus str. Delta H ', '187420-g'),\n",
       "  ('Methanothermococcus okinawensis IH1 ', '647113-g'),\n",
       "  ('Methanothermus fervidus DSM 2088 ', '523846-g'),\n",
       "  ('Methanotorris igneus Kol 5 ', '880724-g'),\n",
       "  ('Nanoarchaeum equitans Kin4-M ', '228908-g'),\n",
       "  ('Natrialba magadii ATCC 43099 ', '547559-g'),\n",
       "  ('Natrinema pellirubrum DSM 15624 ', '797303-g'),\n",
       "  ('Natrinema sp. J7-2 ', '406552-g'),\n",
       "  ('Natronobacterium gregoryi SP2 ', '797304-g'),\n",
       "  ('Natronococcus occultus SP4 ', '694430-g'),\n",
       "  ('Natronomonas moolapensis 8.8.11 ', '268739-g'),\n",
       "  ('Natronomonas pharaonis DSM 2160 ', '348780-g'),\n",
       "  ('Nitrosopumilus maritimus SCM1 ', '436308-g'),\n",
       "  ('Nitrososphaera viennensis EN76 ', '926571-g'),\n",
       "  ('Palaeococcus pacificus DY20341 ', '1343739-g'),\n",
       "  ('Picrophilus torridus DSM 9790 ', '263820-g'),\n",
       "  ('Pyrobaculum aerophilum str. IM2 ', '178306-g'),\n",
       "  ('Pyrobaculum arsenaticum DSM 13514 ', '340102-g'),\n",
       "  ('Pyrobaculum calidifontis JCM 11548 ', '410359-g'),\n",
       "  ('Pyrobaculum islandicum DSM 4184 ', '384616-g'),\n",
       "  ('Pyrobaculum neutrophilum V24Sta ', '444157-g'),\n",
       "  ('Pyrobaculum oguniense TE7 ', '698757-g'),\n",
       "  ('Pyrobaculum sp. 1860 ', '1104324-g'),\n",
       "  ('Pyrococcus abyssi GE5 ', '272844-g'),\n",
       "  ('Pyrococcus furiosus COM1 ', '1185654-g'),\n",
       "  ('Pyrococcus furiosus DSM 3638 ', '186497-g'),\n",
       "  ('Pyrococcus horikoshii OT3 ', '70601-g'),\n",
       "  ('Pyrococcus sp. NA2 ', '342949-g'),\n",
       "  ('Pyrococcus sp. ST04 ', '1183377-g'),\n",
       "  ('Pyrococcus yayanosii CH1 ', '529709-g'),\n",
       "  ('Pyrolobus fumarii 1A ', '694429-g'),\n",
       "  ('Staphylothermus hellenicus DSM 12710 ', '591019-g'),\n",
       "  ('Staphylothermus marinus F1 ', '399550-g'),\n",
       "  ('Sulfolobus acidocaldarius DSM 639 ', '330779-g'),\n",
       "  ('Sulfolobus acidocaldarius N8 ', '1028566-g'),\n",
       "  ('Sulfolobus acidocaldarius Ron12/I ', '1028567-g'),\n",
       "  ('Sulfolobus acidocaldarius SUSAZ ', '1435377-g'),\n",
       "  ('Sulfolobus islandicus HVE10/4 ', '930943-g'),\n",
       "  ('Sulfolobus islandicus L.D.8.5 ', '425944-g'),\n",
       "  ('Sulfolobus islandicus L.S.2.15 ', '429572-g'),\n",
       "  ('Sulfolobus islandicus LAL14/1 ', '1241935-g'),\n",
       "  ('Sulfolobus islandicus M.14.25 ', '427317-g'),\n",
       "  ('Sulfolobus islandicus M.16.27 ', '427318-g'),\n",
       "  ('Sulfolobus islandicus M.16.4 ', '426118-g'),\n",
       "  ('Sulfolobus islandicus REY15A ', '930945-g'),\n",
       "  ('Sulfolobus islandicus Y.G.57.14 ', '439386-g'),\n",
       "  ('Sulfolobus islandicus Y.N.15.51 ', '419942-g'),\n",
       "  ('Sulfolobus solfataricus ', '2287-g'),\n",
       "  ('Sulfolobus solfataricus 98/2 ', '555311-g'),\n",
       "  ('Sulfolobus solfataricus P2 ', '273057-g'),\n",
       "  ('Sulfolobus tokodaii str. 7 ', '273063-g'),\n",
       "  ('Thermococcus barophilus MP ', '391623-g'),\n",
       "  ('Thermococcus cleftensis ', '163003-g'),\n",
       "  ('Thermococcus eurythermalis ', '1505907-g'),\n",
       "  ('Thermococcus gammatolerans EJ3 ', '593117-g'),\n",
       "  ('Thermococcus kodakarensis KOD1 ', '69014-g'),\n",
       "  ('Thermococcus litoralis DSM 5473 ', '523849-g'),\n",
       "  ('Thermococcus nautili ', '195522-g'),\n",
       "  ('Thermococcus onnurineus NA1 ', '523850-g'),\n",
       "  ('Thermococcus paralvinellae ', '582419-g'),\n",
       "  ('Thermococcus sibiricus MM 739 ', '604354-g'),\n",
       "  ('Thermococcus sp. 4557 ', '1042877-g'),\n",
       "  ('Thermococcus sp. AM4 ', '246969-g'),\n",
       "  ('Thermofilum pendens Hrk 5 ', '368408-g'),\n",
       "  ('Thermofilum sp. 1910b ', '1365176-g'),\n",
       "  ('Thermogladius cellulolyticus 1633 ', '1184251-g'),\n",
       "  ('Thermoplasma acidophilum DSM 1728 ', '273075-g'),\n",
       "  ('Thermoplasma volcanium GSS1 ', '273116-g'),\n",
       "  ('Thermoplasmatales archaeon BRNA1 ', '1054217-g'),\n",
       "  ('Thermoproteus tenax Kra 1 ', '768679-g'),\n",
       "  ('Thermoproteus uzoniensis 768-20 ', '999630-g'),\n",
       "  ('Thermosphaera aggregans DSM 11486 ', '633148-g'),\n",
       "  ('Vulcanisaeta distributa DSM 14429 ', '572478-g'),\n",
       "  ('Vulcanisaeta moutnovskia 768-28 ', '985053-g'),\n",
       "  ('uncultured archaeon ', '115547-g'),\n",
       "  (\"'Nostoc azollae' 0708 \", '551115-g'),\n",
       "  ('Acaryochloris marina MBIC11017 ', '329726-g'),\n",
       "  ('Acetobacter pasteurianus 386B ', '1266844-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-01 ', '634452-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-01-42C ', '634458-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-03 ', '634453-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-07 ', '634454-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-12 ', '634459-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-22 ', '634455-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-26 ', '634456-g'),\n",
       "  ('Acetobacter pasteurianus IFO 3283-32 ', '634457-g'),\n",
       "  ('Acetobacterium woodii DSM 1030 ', '931626-g'),\n",
       "  ('Acetohalobium arabaticum DSM 5501 ', '574087-g'),\n",
       "  ('Acholeplasma laidlawii PG-8A ', '441768-g'),\n",
       "  ('Acholeplasma oculi ', '35623-g'),\n",
       "  ('Achromobacter xylosoxidans A8 ', '762376-g'),\n",
       "  ('Achromobacter xylosoxidans C54 ', '562971-g'),\n",
       "  ('Achromobacter xylosoxidans NBRC 15126 = ATCC 27061 ', '1216976-g'),\n",
       "  ('Achromobacter xylosoxidans NH44784-1996 ', '1167634-g'),\n",
       "  ('Acidaminococcus fermentans DSM 20731 ', '591001-g'),\n",
       "  ('Acidaminococcus intestini RyC-MR95 ', '568816-g'),\n",
       "  ('Acidimicrobium ferrooxidans DSM 10331 ', '525909-g'),\n",
       "  ('Acidiphilium cryptum JF-5 ', '349163-g'),\n",
       "  ('Acidiphilium multivorum AIU301 ', '926570-g'),\n",
       "  ('Acidithiobacillus caldus SM-1 ', '990288-g'),\n",
       "  ('Acidithiobacillus ferrivorans SS3 ', '743299-g'),\n",
       "  ('Acidithiobacillus ferrooxidans ATCC 23270 ', '243159-g'),\n",
       "  ('Acidithiobacillus ferrooxidans ATCC 53993 ', '380394-g'),\n",
       "  ('Acidobacterium capsulatum ATCC 51196 ', '240015-g'),\n",
       "  ('Acidothermus cellulolyticus 11B ', '351607-g'),\n",
       "  ('Acidovorax avenae subsp. avenae ATCC 19860 ', '643561-g'),\n",
       "  ('Acidovorax citrulli AAC00-1 ', '397945-g'),\n",
       "  ('Acidovorax ebreus TPSY ', '535289-g'),\n",
       "  ('Acidovorax sp. JS42 ', '232721-g'),\n",
       "  ('Acidovorax sp. KKS102 ', '358220-g'),\n",
       "  ('Acinetobacter baumannii ', '470-g'),\n",
       "  ('Acinetobacter baumannii 1656-2 ', '696749-g'),\n",
       "  ('Acinetobacter baumannii AB0057 ', '480119-g'),\n",
       "  ('Acinetobacter baumannii AB307-0294 ', '557600-g'),\n",
       "  ('Acinetobacter baumannii AC12 ', '1198108-g'),\n",
       "  ('Acinetobacter baumannii AC30 ', '1224747-g'),\n",
       "  ('Acinetobacter baumannii ACICU ', '405416-g'),\n",
       "  ('Acinetobacter baumannii ATCC 17978 ', '400667-g'),\n",
       "  ('Acinetobacter baumannii BJAB07104 ', '1096995-g'),\n",
       "  ('Acinetobacter baumannii BJAB0715 ', '1096996-g'),\n",
       "  ('Acinetobacter baumannii BJAB0868 ', '1096997-g'),\n",
       "  ('Acinetobacter baumannii D1279779 ', '945556-g'),\n",
       "  ('Acinetobacter baumannii LAC-4 ', '1455315-g'),\n",
       "  ('Acinetobacter baumannii MDR-TJ ', '889738-g'),\n",
       "  ('Acinetobacter baumannii MDR-ZJ06 ', '497978-g'),\n",
       "  ('Acinetobacter baumannii NCGM 237 ', '1401639-g'),\n",
       "  ('Acinetobacter baumannii SDF ', '509170-g'),\n",
       "  ('Acinetobacter baumannii TCDC-AB0715 ', '980514-g'),\n",
       "  ('Acinetobacter baumannii TYTH-1 ', '1100841-g'),\n",
       "  ('Acinetobacter baumannii ZW85-1 ', '1400867-g'),\n",
       "  ('Acinetobacter calcoaceticus PHEA-2 ', '871585-g'),\n",
       "  ('Acinetobacter oleivorans DR1 ', '436717-g'),\n",
       "  ('Acinetobacter sp. ADP1 ', '62977-g'),\n",
       "  ('Actinobacillus equuli subsp. equuli ', '202947-g'),\n",
       "  ('Actinobacillus pleuropneumoniae serovar 3 str. JL03 ', '434271-g'),\n",
       "  ('Actinobacillus pleuropneumoniae serovar 5b str. L20 ', '416269-g'),\n",
       "  ('Actinobacillus pleuropneumoniae serovar 7 str. AP76 ', '537457-g'),\n",
       "  ('Actinobacillus succinogenes 130Z ', '339671-g'),\n",
       "  ('Actinobacillus suis ATCC 33415 ', '743972-g'),\n",
       "  ('Actinobacillus suis H91-0380 ', '696748-g'),\n",
       "  ('Actinoplanes friuliensis DSM 7358 ', '1246995-g'),\n",
       "  ('Actinoplanes missouriensis 431 ', '512565-g'),\n",
       "  ...]}"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ews.fasta_search.get_parameter_details('database')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "similar = ews.fasta_search(first_entry.as_fasta())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# /nfs/public/ro/es/appbin-hxwp/linux-x86_64/fasta-36.3.8g/fasta36 -l /nfs/public/ro/es/data/idata/latest/fastacfg/fasta3db -L -T 8 -p -m \"F9 fasta-R20180508-093013-0715-82879346-p2m.m9\" @:1- +uniprotkb_swissprot+\n",
      "FASTA searches a protein or DNA sequence data bank\n",
      " version 36.3.8g Dec, 2017\n",
      "Please cite:\n",
      " W.R. Pearson & D.J. Lipman PNAS (1988) 85:2444-2448\n",
      "\n",
      "Query: @\n",
      "  1>>>sp|Q06507|ATF4_MOUSE Cyclic AMP-dependent transcription factor ATF-4 OS=Mus musculus (Mouse). OX=['10090'] - 349 aa\n",
      "Library: UniProtKB/Swiss-Prot\n",
      "  199856860 residues in 557275 sequences\n",
      "\n",
      "Statistics:  Expectation_n fit: rho(ln(x))= 9.2732+/-0.000157; mu= 0.7113+/- 0.009\n",
      " mean_var=129.2748+/-24.855, 0's: 400 Z-trim(117.9): 511  B-trim: 0 in 0/66\n",
      " Lambda= 0.112802\n",
      " statistics sampled from 60000 (65507) to 100396 sequences\n",
      "Algorithm: FASTA (3.8 Nov 2011) [optimized]\n",
      "Parameters: BL50 matrix (15:-5), open/ext: -10/-2\n",
      " ktup: 2, E-join: 1 (0.507), E-opt: 0.2 (0.181), width:  16\n",
      " Scan time: 11.320\n",
      "\n",
      "The best scores are:                                      opt bits E(557275)\n",
      "SP:ATF4_MOUSE Q06507 Cyclic AMP-dependent transcri ( 349) 2335 391.2 1.1e-107\n",
      "SP:ATF4_RAT Q9ES19 Cyclic AMP-dependent transcript ( 347) 2188 367.3 1.8e-100\n",
      "SP:ATF4_HUMAN P18848 Cyclic AMP-dependent transcri ( 351) 1992 335.4 7.3e-91\n",
      "SP:ATF4_BOVIN Q3ZCH6 Cyclic AMP-dependent transcri ( 348) 1767 298.8 7.6e-80\n",
      "SP:ATF4_DANRE Q6NW59 Cyclic AMP-dependent transcri ( 339)  395 75.5 1.2e-12\n",
      "SP:ATF5_HUMAN Q9Y2D1 Cyclic AMP-dependent transcri ( 282)  320 63.3 4.7e-09\n",
      "SP:ATF5_MOUSE O70191 Cyclic AMP-dependent transcri ( 283)  315 62.5 8.3e-09\n",
      "SP:ATF5_RAT Q\n"
     ]
    }
   ],
   "source": [
    "print(similar.text[:1600])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "similar_ssearch = ews.fasta_search(first_entry.as_fasta(), program='ssearch')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# /nfs/public/ro/es/appbin-hxwp/linux-x86_64/fasta-36.3.8g/ssearch36 -l /nfs/public/ro/es/data/idata/latest/fastacfg/fasta3db -L -T 8 -p -m \"F9 fasta-R20180508-123634-0553-88382806-p1m.m9\" @:1- +uniprotkb_swissprot+\n",
      "SSEARCH performs a Smith-Waterman search\n",
      " version 36.3.8g Dec, 2017\n",
      "Please cite:\n",
      " T. F. Smith and M. S. Waterman, (1981) J. Mol. Biol. 147:195-197; \n",
      " W.R. Pearson (1991) Genomics 11:635-650\n",
      "\n",
      "Query: @\n",
      "  1>>>sp|Q06507|ATF4_MOUSE Cyclic AMP-dependent transcription factor ATF-4 OS=Mus musculus (Mouse). OX=['10090'] - 349 aa\n",
      "Library: UniProtKB/Swiss-Prot\n",
      "  199856860 residues in 557275 sequences\n",
      "\n",
      "Statistics:  Expectation_n fit: rho(ln(x))= 8.9509+/-0.000212; mu= -0.8791+/- 0.012\n",
      " mean_var=112.8684+/-22.192, 0's: 105 Z-trim(131.3): 116  B-trim: 111 in 1/63\n",
      " Lambda= 0.120722\n",
      " statistics sampled from 60000 (193456) to 557071 sequences\n",
      "Algorithm: Smith-Waterman (SSE2, Michael Farrar 2006) (7.2 Nov 2010)\n",
      "Parameters: BL50 matrix (15:-5), open/ext: -10/-2\n",
      " Scan time: 20.360\n",
      "\n",
      "The best scores are:                                      s-w bits E(557275)\n",
      "SP:ATF4_MOUSE Q06507 Cyclic AMP-dependent transcri ( 349) 2335 415.4  6e-115\n",
      "SP:ATF4_RAT Q9ES19 Cyclic AMP-dependent transcript ( 347) 2188 389.8  3e-107\n",
      "SP:ATF4_HUMAN P18848 Cyclic AMP-dependent transcri ( 351) 1992 355.7 5.8e-97\n",
      "SP:ATF4_BOVIN Q3ZCH6 Cyclic AMP-dependent transcri ( 348) 1767 316.5 3.6e-85\n",
      "SP:ATF4_DANRE Q6NW59 Cyclic AMP-dependent transcri ( 339)  544 103.5 4.6e-21\n",
      "SP:ATF5_MOUSE O70191 Cyclic AMP-dependent transcri ( 283)  378 74.6 1.9e-12\n",
      "SP:ATF5_RAT Q6P788 Cyclic AMP-dependent transcript ( 281)  375 74.1 2.7e-\n"
     ]
    }
   ],
   "source": [
    "print(similar_ssearch.text[:1600])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How about using InterPro's HMMER search instead of FASTA?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from prunito import interpro"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "ip_similar = interpro.search_phmmer(first_entry.as_fasta())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "acc2\tacc\tdesc\tspecies\tkg\tevalue\n",
      "Q06507\tATF4_MOUSE\tCyclic AMP-dependent transcription factor ATF-4\tMus musculus\tEukaryota\t1.0e-232\t\n",
      "Q9ES19\tATF4_RAT\tCyclic AMP-dependent transcription factor ATF-4\tRattus norvegicus\tEukaryota\t2.6e-216\t\n",
      "P18848\tATF4_HUMAN\tCyclic AMP-dependent transcription factor ATF-4\tHomo sapiens\tEukaryota\t2.4e-195\t\n",
      "Q3ZCH6\tATF4_BOVIN\tCyclic AMP-dependent transcription factor ATF-4\tBos taurus\tEukaryota\t1.9e-169\t\n",
      "Q6NW59\tATF4_DANRE\tCyclic AMP-dependent transcription factor ATF-4\tDanio rerio\tEukaryota\t5.0e-34\t\n",
      "Q9Y2D1\tATF5_HUMAN\tCyclic AMP-dependent transcription factor ATF-5\tHomo sapiens\tEukaryota\t6.3e-20\t\n",
      "Q6P788\tATF5_RAT\tCyclic AMP-dependent transcription factor ATF-5\tRattus norvegicus\tEukaryota\t5.8e-18\t\n",
      "Q9GPH3\tATFC_BOMMO\tActivating transcription factor of chaperone\tBombyx mori\tEukaryota\t2.4e-16\t\n",
      "O70191\tATF5_MOUSE\tCyclic AMP-dependent transcription factor ATF-5\tMus musculus\tEukaryota\t3.5e-13\t\n",
      "Q8TFF3\tHAC1_HYPJE\tTranscriptional activator hac1\tHypocrea jecorina (strain QM6a)\tEukaryota\t5.4e-05\t\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(ip_similar.summary())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_hmmer = ip_similar.as_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>acc2</th>\n",
       "      <th>acc</th>\n",
       "      <th>desc</th>\n",
       "      <th>species</th>\n",
       "      <th>kg</th>\n",
       "      <th>evalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Q06507</th>\n",
       "      <td>ATF4_MOUSE</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-4</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>1.000000e-232</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9ES19</th>\n",
       "      <td>ATF4_RAT</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-4</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>2.600000e-216</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P18848</th>\n",
       "      <td>ATF4_HUMAN</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-4</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>2.400000e-195</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q3ZCH6</th>\n",
       "      <td>ATF4_BOVIN</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-4</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>1.900000e-169</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q6NW59</th>\n",
       "      <td>ATF4_DANRE</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-4</td>\n",
       "      <td>Danio rerio</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>5.000000e-34</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9Y2D1</th>\n",
       "      <td>ATF5_HUMAN</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-5</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>6.300000e-20</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q6P788</th>\n",
       "      <td>ATF5_RAT</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-5</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>5.800000e-18</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q9GPH3</th>\n",
       "      <td>ATFC_BOMMO</td>\n",
       "      <td>Activating transcription factor of chaperone</td>\n",
       "      <td>Bombyx mori</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>2.400000e-16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>O70191</th>\n",
       "      <td>ATF5_MOUSE</td>\n",
       "      <td>Cyclic AMP-dependent transcription factor ATF-5</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>3.500000e-13</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q8TFF3</th>\n",
       "      <td>HAC1_HYPJE</td>\n",
       "      <td>Transcriptional activator hac1</td>\n",
       "      <td>Hypocrea jecorina (strain QM6a)</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>5.400000e-05</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              acc2                                              acc  \\\n",
       "Q06507  ATF4_MOUSE  Cyclic AMP-dependent transcription factor ATF-4   \n",
       "Q9ES19    ATF4_RAT  Cyclic AMP-dependent transcription factor ATF-4   \n",
       "P18848  ATF4_HUMAN  Cyclic AMP-dependent transcription factor ATF-4   \n",
       "Q3ZCH6  ATF4_BOVIN  Cyclic AMP-dependent transcription factor ATF-4   \n",
       "Q6NW59  ATF4_DANRE  Cyclic AMP-dependent transcription factor ATF-4   \n",
       "Q9Y2D1  ATF5_HUMAN  Cyclic AMP-dependent transcription factor ATF-5   \n",
       "Q6P788    ATF5_RAT  Cyclic AMP-dependent transcription factor ATF-5   \n",
       "Q9GPH3  ATFC_BOMMO     Activating transcription factor of chaperone   \n",
       "O70191  ATF5_MOUSE  Cyclic AMP-dependent transcription factor ATF-5   \n",
       "Q8TFF3  HAC1_HYPJE                   Transcriptional activator hac1   \n",
       "\n",
       "                                   desc    species             kg  evalue  \n",
       "Q06507                     Mus musculus  Eukaryota  1.000000e-232     NaN  \n",
       "Q9ES19                Rattus norvegicus  Eukaryota  2.600000e-216     NaN  \n",
       "P18848                     Homo sapiens  Eukaryota  2.400000e-195     NaN  \n",
       "Q3ZCH6                       Bos taurus  Eukaryota  1.900000e-169     NaN  \n",
       "Q6NW59                      Danio rerio  Eukaryota   5.000000e-34     NaN  \n",
       "Q9Y2D1                     Homo sapiens  Eukaryota   6.300000e-20     NaN  \n",
       "Q6P788                Rattus norvegicus  Eukaryota   5.800000e-18     NaN  \n",
       "Q9GPH3                      Bombyx mori  Eukaryota   2.400000e-16     NaN  \n",
       "O70191                     Mus musculus  Eukaryota   3.500000e-13     NaN  \n",
       "Q8TFF3  Hypocrea jecorina (strain QM6a)  Eukaryota   5.400000e-05     NaN  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_hmmer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "vn = interpro.draw_signature_overlaps(['MF_02067', 'PTHR38341', 'PIRSF015647', 'PF01950'], mode='ipython')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<svg height=\"503.07043\" id=\"svg2\" inkscape:version=\"0.92.0 r15299\" sodipodi:docname=\"4_set.svg\" version=\"1.1\" width=\"677.08765\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:inkscape=\"http://www.inkscape.org/namespaces/inkscape\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" xmlns:sodipodi=\"http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd\" xmlns:svg=\"http://www.w3.org/2000/svg\">\n",
       "  <defs id=\"defs4\"/>\n",
       "  <sodipodi:namedview bordercolor=\"#666666\" borderopacity=\"1.0\" fit-margin-bottom=\"35\" fit-margin-left=\"20\" fit-margin-right=\"20\" fit-margin-top=\"35\" id=\"base\" inkscape:current-layer=\"layer8\" inkscape:cx=\"412.16645\" inkscape:cy=\"208.25143\" inkscape:document-units=\"px\" inkscape:guide-bbox=\"true\" inkscape:pageopacity=\"0.0\" inkscape:pageshadow=\"2\" inkscape:showpageshadow=\"true\" inkscape:window-height=\"1017\" inkscape:window-maximized=\"1\" inkscape:window-width=\"1920\" inkscape:window-x=\"-8\" inkscape:window-y=\"-8\" inkscape:zoom=\"1.4\" pagecolor=\"#ffffff\" showgrid=\"false\" showguides=\"false\">\n",
       "    <sodipodi:guide id=\"guide3781\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"-331.11998,194.96663\"/>\n",
       "    <sodipodi:guide id=\"guide3783\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"331.5401,194.96663\"/>\n",
       "    <sodipodi:guide id=\"guide3785\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"186.07812,-198.85817\"/>\n",
       "    <sodipodi:guide id=\"guide3787\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"186.07812,-198.85817\"/>\n",
       "    <sodipodi:guide id=\"guide3789\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"273.99293,214.90642\"/>\n",
       "    <sodipodi:guide id=\"guide3791\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"976.96027,135.16561\"/>\n",
       "    <sodipodi:guide id=\"guide3793\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"331.5401,250.45769\"/>\n",
       "    <sodipodi:guide id=\"guide3795\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"955.41029,-198.85817\"/>\n",
       "    <sodipodi:guide id=\"guide3799\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"1007.6689,-1.137614\"/>\n",
       "    <sodipodi:guide id=\"guide3801\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"1054.0012,-28.075014\"/>\n",
       "    <sodipodi:guide id=\"guide3803\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"1007.6689,-164.37829\"/>\n",
       "    <sodipodi:guide id=\"guide3900\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"-358.05738,219.2103\"/>\n",
       "    <sodipodi:guide id=\"guide3902\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"388.27863,219.85879\"/>\n",
       "    <sodipodi:guide id=\"guide3904\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"197.04054,307.47785\"/>\n",
       "    <sodipodi:guide id=\"guide3906\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"202.37388,194.33499\"/>\n",
       "    <sodipodi:guide id=\"guide3908\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"230.9453,323.85879\"/>\n",
       "    <sodipodi:guide id=\"guide3910\" inkscape:locked=\"false\" orientation=\"1,0\" position=\"429.04054,307.47785\"/>\n",
       "    <sodipodi:guide id=\"guide3912\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"331.5401,349.00165\"/>\n",
       "    <sodipodi:guide id=\"guide3914\" inkscape:locked=\"false\" orientation=\"0,1\" position=\"331.5401,156.23975\"/>\n",
       "  </sodipodi:namedview>\n",
       "  <metadata id=\"metadata7\">\n",
       "    <rdf:RDF>\n",
       "      <cc:Work rdf:about=\"\">\n",
       "        <dc:format>image/svg+xml</dc:format>\n",
       "        <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "        <dc:title/>\n",
       "      </cc:Work>\n",
       "    </rdf:RDF>\n",
       "  </metadata>\n",
       "  <g id=\"layer1\" inkscape:groupmode=\"layer\" inkscape:label=\"2-set\" style=\"display:inline\" transform=\"translate(-277.24517,-501.40339)\">\n",
       "    <rect height=\"212.088\" id=\"rect2985\" style=\"opacity:0.4;fill:#e41a1c;fill-opacity:1;stroke:#000000;stroke-width:0.81655955;stroke-opacity:1\" width=\"633.60443\" x=\"298.98679\" y=\"754.64423\"/>\n",
       "    <rect height=\"427.57675\" id=\"rect3755\" style=\"opacity:0.4;fill:#377eb8;fill-opacity:1;stroke:#000000;stroke-width:0.82688004;stroke-linecap:round;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1\" width=\"322.41306\" x=\"610.17285\" y=\"539.15015\"/>\n",
       "  </g>\n",
       "  <g id=\"layer2\" inkscape:groupmode=\"layer\" inkscape:label=\"2-set-labels\" style=\"display:none\">\n",
       "    <text id=\"text3054\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"26.66667\" xml:space=\"preserve\" y=\"458.87997\"><tspan id=\"tspan3056\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"26.66667\" y=\"458.87997\">10</tspan></text>\n",
       "    <text id=\"text3058\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"617.14282\" xml:space=\"preserve\" y=\"52.784733\"><tspan id=\"tspan3060\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"617.14282\" y=\"52.784733\">1</tspan></text>\n",
       "    <text id=\"text3062\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"620.95239\" xml:space=\"preserve\" y=\"457.35617\"><tspan id=\"tspan3064\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"620.95239\" y=\"457.35617\">11</tspan></text>\n",
       "    <text id=\"text3116\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"30.476191\" xml:space=\"preserve\" y=\"240.2133\"><tspan id=\"tspan3118\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"30.476191\" y=\"240.2133\">MF_02067</tspan></text>\n",
       "    <text id=\"text3120\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"270.4762\" xml:space=\"preserve\" y=\"59.641869\"><tspan id=\"tspan3122\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"270.4762\" y=\"59.641869\">PTHR38341</tspan></text>\n",
       "  </g>\n",
       "  <g id=\"layer3\" inkscape:groupmode=\"layer\" inkscape:label=\"3-set\" style=\"display:inline\" transform=\"translate(-277.24517,-172.58451)\">\n",
       "    <path d=\"M 208.96337,183.80998 141.06732,235.90845 56.218383,224.73787 4.1199123,156.84182 15.290487,71.992881 83.186538,19.89441 l 84.848942,11.170575 52.09847,67.89605 z\" id=\"path3898\" inkscape:flatsided=\"true\" inkscape:randomized=\"0\" inkscape:rounded=\"0\" sodipodi:arg1=\"0.52359878\" sodipodi:arg2=\"0.91629786\" sodipodi:cx=\"112.12693\" sodipodi:cy=\"127.90143\" sodipodi:r1=\"111.81709\" sodipodi:r2=\"103.30552\" sodipodi:sides=\"8\" sodipodi:type=\"star\" style=\"opacity:0.45;fill:#4daf4a;fill-opacity:1;stroke:#000000;stroke-width:1;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1\" transform=\"matrix(1.0515685,-0.13497884,0.13497884,1.0515685,474.97822,305.58465)\"/>\n",
       "  </g>\n",
       "  <g id=\"layer7\" inkscape:groupmode=\"layer\" inkscape:label=\"3-set-labels\" style=\"display:none\">\n",
       "    <text id=\"text3054-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"28.023502\" xml:space=\"preserve\" y=\"456.89783\"><tspan id=\"tspan3056-7\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"28.023502\" y=\"456.89783\">100</tspan></text>\n",
       "    <text id=\"text3058-4\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"618.49969\" xml:space=\"preserve\" y=\"50.802578\"><tspan id=\"tspan3060-0\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"618.49969\" y=\"50.802578\">10</tspan></text>\n",
       "    <text id=\"text3062-9\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"622.30927\" xml:space=\"preserve\" y=\"455.37402\"><tspan id=\"tspan3064-4\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"622.30927\" y=\"455.37402\">110</tspan></text>\n",
       "    <text id=\"text3100\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"271.07114\" xml:space=\"preserve\" y=\"210.0407\"><tspan id=\"tspan3102\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"271.07114\" y=\"210.0407\">1</tspan></text>\n",
       "    <text id=\"text3104\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"271.07114\" xml:space=\"preserve\" y=\"293.24069\"><tspan id=\"tspan3106\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"271.07114\" y=\"293.24069\">101</tspan></text>\n",
       "    <text id=\"text3108\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"364.93784\" xml:space=\"preserve\" y=\"293.24069\"><tspan id=\"tspan3110\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"364.93784\" y=\"293.24069\">111</tspan></text>\n",
       "    <text id=\"text3112\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"364.93784\" xml:space=\"preserve\" y=\"207.90736\"><tspan id=\"tspan3114\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"364.93784\" y=\"207.90736\">11</tspan></text>\n",
       "    <text id=\"text3116-8\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"32.523033\" xml:space=\"preserve\" y=\"239.25795\"><tspan id=\"tspan3118-8\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"32.523033\" y=\"239.25795\">MF_02067</tspan></text>\n",
       "    <text id=\"text3120-2\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"272.52304\" xml:space=\"preserve\" y=\"58.686527\"><tspan id=\"tspan3122-4\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"272.52304\" y=\"58.686527\">PTHR38341</tspan></text>\n",
       "    <text id=\"text3151\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"189.4754\" xml:space=\"preserve\" y=\"170.68652\"><tspan id=\"tspan3153\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"189.4754\" y=\"170.68652\">PIRSF015647</tspan></text>\n",
       "  </g>\n",
       "  <g id=\"layer4\" inkscape:groupmode=\"layer\" inkscape:label=\"4-set\" style=\"display:inline\" transform=\"translate(-277.24517,-172.58451)\">\n",
       "    <path d=\"m 328.40905,297.50211 1.21905,251.00614 139.43446,-0.20835 95.08572,-78.42009 h 91.42856 l 99.35239,78.47308 143.92972,0.15536 -0.60952,-251.23131 -143.3202,0.8998 -99.35239,81.18877 h -91.42856 l -95.08572,-81.01499 z\" id=\"path3767\" inkscape:connector-curvature=\"0\" sodipodi:nodetypes=\"ccccccccccccc\" style=\"display:inline;opacity:0.45;fill:#984ea3;fill-opacity:1;stroke:#000000;stroke-width:1.04532552px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1\"/>\n",
       "  </g>\n",
       "  <g id=\"layer8\" inkscape:groupmode=\"layer\" inkscape:label=\"4-set-labels\" style=\"display:inline\">\n",
       "    <text id=\"text3054-1-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"168.04851\" xml:space=\"preserve\" y=\"419.66284\"><tspan id=\"tspan3056-7-7\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"168.04851\" y=\"419.66284\">0</tspan></text>\n",
       "    <text id=\"text3058-4-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"483.52469\" xml:space=\"preserve\" y=\"86.424744\"><tspan id=\"tspan3060-0-1\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"483.52469\" y=\"86.424744\">0</tspan></text>\n",
       "    <text id=\"text3062-9-5\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"498.04855\" xml:space=\"preserve\" y=\"413.13904\"><tspan id=\"tspan3064-4-2\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"498.04855\" y=\"413.13904\">0</tspan></text>\n",
       "    <text id=\"text3100-7\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"299.09613\" xml:space=\"preserve\" y=\"179.90097\"><tspan id=\"tspan3102-6\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"299.09613\" y=\"179.90097\">0</tspan></text>\n",
       "    <text id=\"text3104-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"297.09613\" xml:space=\"preserve\" y=\"330.24377\"><tspan id=\"tspan3106-4\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"297.09613\" y=\"330.24377\">0</tspan></text>\n",
       "    <text id=\"text3108-2\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"362.81995\" xml:space=\"preserve\" y=\"331.57712\"><tspan id=\"tspan3110-3\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"362.81995\" y=\"331.57712\">0</tspan></text>\n",
       "    <text id=\"text3112-2\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"360.01044\" xml:space=\"preserve\" y=\"179.29144\"><tspan id=\"tspan3114-2\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"360.01044\" y=\"179.29144\">0</tspan></text>\n",
       "    <text id=\"text3116-8-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"25.786127\" xml:space=\"preserve\" y=\"455.1658\"><tspan id=\"tspan3118-8-6\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"25.786127\" y=\"455.1658\">MF_02067</tspan></text>\n",
       "    <text id=\"text3120-2-8\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"340.78613\" xml:space=\"preserve\" y=\"61.642021\"><tspan id=\"tspan3122-4-5\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"340.78613\" y=\"61.642021\">PTHR38341</tspan></text>\n",
       "    <text id=\"text3151-7\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"259.40515\" xml:space=\"preserve\" y=\"159.0706\"><tspan id=\"tspan3153-6\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"259.40515\" y=\"159.0706\">PIRSF015647</tspan></text>\n",
       "    <text id=\"text3252\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"56.548031\" xml:space=\"preserve\" y=\"145.07057\"><tspan id=\"tspan3254\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"56.548031\" y=\"145.07057\">PF01950</tspan></text>\n",
       "    <text id=\"text3260\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"125.06757\" xml:space=\"preserve\" y=\"191.32953\"><tspan id=\"tspan3262\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"125.06757\" y=\"191.32953\">12</tspan></text>\n",
       "    <text id=\"text3264\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"127.71519\" xml:space=\"preserve\" y=\"319.48187\"><tspan id=\"tspan3266\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"127.71519\" y=\"319.48187\">0</tspan></text>\n",
       "    <text id=\"text3268\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"520.49609\" xml:space=\"preserve\" y=\"324.48187\"><tspan id=\"tspan3270\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"520.49609\" y=\"324.48187\">36</tspan></text>\n",
       "    <text id=\"text3272\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"517.63898\" xml:space=\"preserve\" y=\"192.97711\"><tspan id=\"tspan3274\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"517.63898\" y=\"192.97711\">171</tspan></text>\n",
       "    <text id=\"text3276\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"379.01993\" xml:space=\"preserve\" y=\"234.04378\"><tspan id=\"tspan3278\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"379.01993\" y=\"234.04378\">13</tspan></text>\n",
       "    <text id=\"text3280\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"271.8009\" xml:space=\"preserve\" y=\"233.48187\"><tspan id=\"tspan3282\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"271.8009\" y=\"233.48187\">1</tspan></text>\n",
       "    <text id=\"text3284\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"378.30566\" xml:space=\"preserve\" y=\"280.2533\"><tspan id=\"tspan3286\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"378.30566\" y=\"280.2533\">714</tspan></text>\n",
       "    <text id=\"text3288\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"270.37231\" xml:space=\"preserve\" y=\"277.539\"><tspan id=\"tspan3290\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"270.37231\" y=\"277.539\">1</tspan></text>\n",
       "  </g>\n",
       "  <g id=\"layer5\" inkscape:groupmode=\"layer\" inkscape:label=\"5-set\" style=\"display:none\" transform=\"translate(-277.24517,-172.58451)\">\n",
       "    <path d=\"m 655.57684,250.54169 -91.42856,-0.65538 v 129.4792 H 441.40675 v 90.5143 h 122.74153 v 128.91748 h 91.42856 V 469.87981 h 136.82679 l -0.30476,-90.5143 H 655.57684 Z\" id=\"path3779\" inkscape:connector-curvature=\"0\" sodipodi:nodetypes=\"ccccccccccccc\" style=\"display:inline;opacity:0.3;fill:#ff7f00;fill-opacity:1;stroke:#000000;stroke-width:1.3363812px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1\"/>\n",
       "  </g>\n",
       "  <g id=\"layer9\" inkscape:groupmode=\"layer\" inkscape:label=\"5-set-lables\" style=\"display:none\">\n",
       "    <text id=\"text3054-1-1-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"23.969429\" xml:space=\"preserve\" y=\"457.43115\"><tspan id=\"tspan3056-7-7-8\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"23.969429\" y=\"457.43115\">10000</tspan></text>\n",
       "    <text id=\"text3058-4-1-9\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"614.44556\" xml:space=\"preserve\" y=\"51.335915\"><tspan id=\"tspan3060-0-1-2\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"614.44556\" y=\"51.335915\">1000</tspan></text>\n",
       "    <text id=\"text3062-9-5-7\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"613.98846\" xml:space=\"preserve\" y=\"455.90735\"><tspan id=\"tspan3064-4-2-9\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"613.98846\" y=\"455.90735\">11000</tspan></text>\n",
       "    <text id=\"text3100-7-5\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"254.82658\" xml:space=\"preserve\" y=\"180.09784\"><tspan id=\"tspan3102-6-4\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"254.82658\" y=\"180.09784\">100</tspan></text>\n",
       "    <text id=\"text3104-1-3\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"252.54086\" xml:space=\"preserve\" y=\"331.10736\"><tspan id=\"tspan3106-4-1\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"252.54086\" y=\"331.10736\">10100</tspan></text>\n",
       "    <text id=\"text3108-2-2\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"374.59802\" xml:space=\"preserve\" y=\"329.58356\"><tspan id=\"tspan3110-3-3\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"374.59802\" y=\"329.58356\">11100</tspan></text>\n",
       "    <text id=\"text3112-2-3\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"375.35992\" xml:space=\"preserve\" y=\"180.25024\"><tspan id=\"tspan3114-2-4\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"375.35992\" y=\"180.25024\">1100</tspan></text>\n",
       "    <text id=\"text3116-8-1-1\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"23.135624\" xml:space=\"preserve\" y=\"484.3627\"><tspan id=\"tspan3118-8-6-1\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"23.135624\" y=\"484.3627\">MF_02067</tspan></text>\n",
       "    <text id=\"text3120-2-8-3\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"598.37372\" xml:space=\"preserve\" y=\"27.98177\"><tspan id=\"tspan3122-4-5-8\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"598.37372\" y=\"27.98177\">PTHR38341</tspan></text>\n",
       "    <text id=\"text3151-7-7\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"218.94513\" xml:space=\"preserve\" y=\"139.98177\"><tspan id=\"tspan3153-6-4\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"218.94513\" y=\"139.98177\">PIRSF015647</tspan></text>\n",
       "    <text id=\"text3252-2\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"58.183243\" xml:space=\"preserve\" y=\"117.1246\"><tspan id=\"tspan3254-7\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"58.183243\" y=\"117.1246\">PF01950</tspan></text>\n",
       "    <text id=\"text3260-7\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"63.131355\" xml:space=\"preserve\" y=\"186.95499\"><tspan id=\"tspan3262-9\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"63.131355\" y=\"186.95499\">10</tspan></text>\n",
       "    <text id=\"text3264-3\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"64.350395\" xml:space=\"preserve\" y=\"326.53592\"><tspan id=\"tspan3266-1\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"64.350395\" y=\"326.53592\">0</tspan></text>\n",
       "    <text id=\"text3268-9\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"561.41705\" xml:space=\"preserve\" y=\"326.53592\"><tspan id=\"tspan3270-8\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"561.41705\" y=\"326.53592\">11010</tspan></text>\n",
       "    <text id=\"text3272-6\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"561.41705\" xml:space=\"preserve\" y=\"183.60257\"><tspan id=\"tspan3274-5\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"561.41705\" y=\"183.60257\">1010</tspan></text>\n",
       "    <text id=\"text3276-0\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"389.22656\" xml:space=\"preserve\" y=\"202.95497\"><tspan id=\"tspan3278-2\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"389.22656\" y=\"202.95497\">1110</tspan></text>\n",
       "    <text id=\"text3280-8\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"235.93134\" xml:space=\"preserve\" y=\"201.58353\"><tspan id=\"tspan3282-6\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"235.93134\" y=\"201.58353\">110</tspan></text>\n",
       "    <text id=\"text3284-0\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"392.2742\" xml:space=\"preserve\" y=\"307.64066\"><tspan id=\"tspan3286-2\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"392.2742\" y=\"307.64066\">11110</tspan></text>\n",
       "    <text id=\"text3288-4\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"235.62657\" xml:space=\"preserve\" y=\"306.87878\"><tspan id=\"tspan3290-8\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"235.62657\" y=\"306.87878\">10110</tspan></text>\n",
       "    <text id=\"text3422\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"253.99275\" xml:space=\"preserve\" y=\"66.838913\"><tspan id=\"tspan3424\" sodipodi:role=\"line\" style=\"font-size:21.33333397px;stroke-width:1.06666672\" x=\"253.99275\" y=\"66.838913\">None</tspan></text>\n",
       "    <text id=\"text3426\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"289.41705\" xml:space=\"preserve\" y=\"92.47879\"><tspan id=\"tspan3428\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"289.41705\" y=\"92.47879\">1</tspan></text>\n",
       "    <text id=\"text3430\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"336.3504\" xml:space=\"preserve\" y=\"92.47879\"><tspan id=\"tspan3432\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"336.3504\" y=\"92.47879\">1001</tspan></text>\n",
       "    <text id=\"text3434\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"291.09323\" xml:space=\"preserve\" y=\"160.89784\"><tspan id=\"tspan3436\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"291.09323\" y=\"160.89784\">101</tspan></text>\n",
       "    <text id=\"text3438\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"340.15988\" xml:space=\"preserve\" y=\"160.89784\"><tspan id=\"tspan3440\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"340.15988\" y=\"160.89784\">1101</tspan></text>\n",
       "    <text id=\"text3442\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"171.62656\" xml:space=\"preserve\" y=\"231.29784\"><tspan id=\"tspan3444\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"171.62656\" y=\"231.29784\">11</tspan></text>\n",
       "    <text id=\"text3446\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"171.62656\" xml:space=\"preserve\" y=\"278.23117\"><tspan id=\"tspan3448\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"171.62656\" y=\"278.23117\">10011</tspan></text>\n",
       "    <text id=\"text3450\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"361.49323\" xml:space=\"preserve\" y=\"232.82162\"><tspan id=\"tspan3452\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"361.49323\" y=\"232.82162\">1111</tspan></text>\n",
       "    <text id=\"text3454\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"361.49323\" xml:space=\"preserve\" y=\"275.48828\"><tspan id=\"tspan3456\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"361.49323\" y=\"275.48828\">11111</tspan></text>\n",
       "    <text id=\"text3458\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"252.69322\" xml:space=\"preserve\" y=\"275.48828\"><tspan id=\"tspan3460\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"252.69322\" y=\"275.48828\">10111</tspan></text>\n",
       "    <text id=\"text3462\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"252.69322\" xml:space=\"preserve\" y=\"232.82161\"><tspan id=\"tspan3464\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"252.69322\" y=\"232.82161\">111</tspan></text>\n",
       "    <text id=\"text3466\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"288.80756\" xml:space=\"preserve\" y=\"348.17401\"><tspan id=\"tspan3468\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"288.80756\" y=\"348.17401\">10101</tspan></text>\n",
       "    <text id=\"text3470\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"337.87421\" xml:space=\"preserve\" y=\"348.17401\"><tspan id=\"tspan3472\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"337.87421\" y=\"348.17401\">1101</tspan></text>\n",
       "    <text id=\"text3474\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"288.80756\" xml:space=\"preserve\" y=\"399.37402\"><tspan id=\"tspan3476\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"288.80756\" y=\"399.37402\">10001</tspan></text>\n",
       "    <text id=\"text3478\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"337.87421\" xml:space=\"preserve\" y=\"399.37402\"><tspan id=\"tspan3480\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"337.87421\" y=\"399.37402\">1001</tspan></text>\n",
       "    <text id=\"text3482\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"455.35989\" xml:space=\"preserve\" y=\"232.82162\"><tspan id=\"tspan3484\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"455.35989\" y=\"232.82162\">1011</tspan></text>\n",
       "    <text id=\"text3486\" style=\"font-style:normal;font-weight:normal;font-size:42.66666794px;line-height:125%;font-family:Sans;letter-spacing:0px;word-spacing:0px;display:inline;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1.06666672\" x=\"455.35989\" xml:space=\"preserve\" y=\"275.48828\"><tspan id=\"tspan3488\" sodipodi:role=\"line\" style=\"font-size:8.53333378px;stroke-width:1.06666672\" x=\"455.35989\" y=\"275.48828\">11011</tspan></text>\n",
       "  </g>\n",
       "  <g id=\"layer6\" inkscape:groupmode=\"layer\" inkscape:label=\"6-set\" style=\"display:none\" transform=\"translate(-277.24517,-172.58451)\">\n",
       "    <path d=\"m 591.38222,292.59843 43.14761,0.41577 21.04701,53.13227 32.60953,-32.05474 33.72932,32.05474 -33.72932,33.21904 67.4211,22.73024 0.0151,45.56169 -67.43627,22.22237 32.80629,30.47618 -32.80629,30.7974 -32.60953,-30.7974 -20.15455,55.83444 h -50.05888 l -21.21513,-55.83444 -34.43811,33.01333 -32.69032,-33.01333 32.69032,-30.47618 -55.58765,-23.9616 v -43.39146 l 55.58765,-23.16124 -33.42616,-33.21904 33.42616,-31.54322 34.43811,31.54322 z\" id=\"path3815\" inkscape:connector-curvature=\"0\" sodipodi:nodetypes=\"ccccccccccccccccccccccccc\" style=\"display:inline;opacity:0.3;fill:#ffff33;fill-opacity:1;stroke:#000000;stroke-width:0.85333335\"/>\n",
       "  </g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.SVG object>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Do some of the 41 entries contain the same PubMed IDs? Let's find the 5 most common ones."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('15489334', 24), ('20068231', 9), ('14702039', 8), ('23186163', 8), ('21269460', 7)]\n"
     ]
    }
   ],
   "source": [
    "c = Counter()\n",
    "for e in entries:\n",
    "    c.update(e.all_pubmed_ids)\n",
    "print(c.most_common(5))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Which are the accession numbers and species of those 24 entries containing the most common one (15489334)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q06507 Mus musculus (Mouse).\n",
      "P18848 Homo sapiens (Human).\n",
      "Q9Y6D9 Homo sapiens (Human).\n",
      "P47911 Mus musculus (Mouse).\n",
      "Q0VGT2 Mus musculus (Mouse).\n",
      "O14908 Homo sapiens (Human).\n",
      "Q02878 Homo sapiens (Human).\n",
      "O14910 Homo sapiens (Human).\n",
      "Q10586 Homo sapiens (Human).\n",
      "Q61330 Mus musculus (Mouse).\n",
      "P43489 Homo sapiens (Human).\n",
      "Q08AM6 Homo sapiens (Human).\n",
      "P23510 Homo sapiens (Human).\n",
      "Q86VP1 Homo sapiens (Human).\n",
      "O35426 Mus musculus (Mouse).\n",
      "P17861 Homo sapiens (Human).\n",
      "Q9DBG9 Mus musculus (Mouse).\n",
      "Q4QQV1 Rattus norvegicus (Rat).\n",
      "O14907 Homo sapiens (Human).\n",
      "Q66HA4 Rattus norvegicus (Rat).\n",
      "Q9UQ35 Homo sapiens (Human).\n",
      "Q13884 Homo sapiens (Human).\n",
      "Q3UKC1 Mus musculus (Mouse).\n",
      "Q9NPB6 Homo sapiens (Human).\n"
     ]
    }
   ],
   "source": [
    "for e in entries:\n",
    "    if '15489334' in e.all_pubmed_ids:\n",
    "        print(e.primary_accession, e.organism)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So, which paper is hiding behind this PMID 15489334? Here we use another module from prunito."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "from prunito import europepmc as epmc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "paper = epmc.get_pmid_metadata('15489334')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "EuropePMC returns data for example in JSON format. We can iterate over results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The status, quality, and expansion of the NIH full-length cDNA project: the Mammalian Gene Collection (MGC).\n"
     ]
    }
   ],
   "source": [
    "# paper.json()['resultList']['result'][0]['title']\n",
    "for p in paper:\n",
    "    print(p['title'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"The National Institutes of Health's Mammalian Gene Collection (MGC) project was designed to generate and sequence a publicly accessible cDNA resource containing a complete open reading frame (ORF) for every human and mouse gene. The project initially used a random strategy to select clones from a large number of cDNA libraries from diverse tissues. Candidate clones were chosen based on 5'-EST sequences, and then fully sequenced to high accuracy and analyzed by algorithms developed for this project. Currently, more than 11,000 human and 10,000 mouse genes are represented in MGC by at least one clone with a full ORF. The random selection approach is now reaching a saturation point, and a transition to protocols targeted at the missing transcripts is now required to complete the mouse and human collections. Comparison of the sequence of the MGC clones to reference genome sequences reveals that most cDNA clones are of very high sequence quality, although it is likely that some cDNAs may carry missense variants as a consequence of experimental artifact, such as PCR, cloning, or reverse transcriptase errors. Recently, a rat cDNA component was added to the project, and ongoing frog (Xenopus) and zebrafish (Danio) cDNA projects were expanded to take advantage of the high-throughput MGC pipeline.\""
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(paper)[0]['abstractText']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As the paper mentions the Mammalian Gene Collection, why not search EuropePMC for articles mentioning the collection in their abstracts?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "mgc_papers = epmc.search('abstract:\"Mammalian Gene Collection\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgc_papers.size()\n",
    "#\n",
    "# len(mgc_papers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Identification of candidate transcription factor binding sites in the cattle genome.\n",
      "1 Selenoproteins in bladder cancer.\n",
      "2 NSrp70 is a novel nuclear speckle-related protein that modulates alternative pre-mRNA splicing in vivo.\n",
      "3 Generation of a genome scale lentiviral vector library for EF1α promoter-driven expression of human ORFs and identification of human genes affecting viral titer.\n",
      "4 The completion of the Mammalian Gene Collection (MGC).\n",
      "5 A high-throughput platform for lentiviral overexpression screening of the human ORFeome.\n",
      "6 PRFdb: a database of computationally predicted eukaryotic programmed -1 ribosomal frameshift signals.\n",
      "7 Transcriptome analysis of a cDNA library from adult human epididymis.\n",
      "8 Construction and characterization of a normalized yeast two-hybrid library derived from a human protein-coding clone collection.\n",
      "9 Evaluation of vector-primed cDNA library production from microgram quantities of total RNA.\n",
      "10 Concatenation cDNA sequencing for transcriptome analysis.\n",
      "11 Targeted discovery of novel human exons by comparative genomics.\n",
      "12 1274 full-open reading frames of transcripts expressed in the developing mouse nervous system.\n",
      "13 The status, quality, and expansion of the NIH full-length cDNA project: the Mammalian Gene Collection (MGC).\n",
      "14 Generation and initial analysis of more than 15,000 full-length human and mouse cDNA sequences.\n",
      "15 Identification and characterisation of human apoptosis inducing proteins using cell-based transfection microarrays and expression analysis.\n",
      "16 The mammalian gene collection.\n",
      "17 Human ORFeome version 1.1: a platform for reverse proteomics.\n",
      "18 Identification and functional analysis of human transcriptional promoters.\n",
      "19 Analysis of human mRNAs with the reference genome sequence reveals potential errors, polymorphisms, and RNA editing.\n",
      "20 3'-UTR SIRF: a database for identifying clusters of whort interspersed repeats in 3' untranslated regions.\n",
      "21 An efficient strategy for large-scale high-throughput transposon-mediated sequencing of cDNA clones.\n",
      "22 Genome and genetic resources from the Cancer Genome Anatomy Project.\n",
      "23 Systematic recovery and analysis of full-ORF human cDNA clones.\n"
     ]
    }
   ],
   "source": [
    "for idx, hit in enumerate(mgc_papers):\n",
    "    print(idx, hit['title'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each hit/paper has many extra data fields including DOI, PubMed ID etc. If the abstract is needed, resulttype='core' has to be specified as a search parameter."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id:\t23251614\n",
      "source:\tMED\n",
      "pmid:\t23251614\n",
      "pmcid:\tPMC3520899\n",
      "doi:\t10.1371/journal.pone.0051733\n",
      "title:\tGeneration of a genome scale lentiviral vector library for EF1α promoter-driven expression of human ORFs and identification of human genes affecting viral titer.\n",
      "authorString:\tŠkalamera D, Dahmer M, Purdon AS, Wilson BM, Ranall MV, Blumenthal A, Gabrielli B, Gonda TJ.\n",
      "journalTitle:\tPLoS One\n",
      "issue:\t12\n",
      "journalVolume:\t7\n",
      "pubYear:\t2012\n",
      "journalIssn:\t1932-6203\n",
      "pageInfo:\te51733\n",
      "pubType:\tresearch support, non-u.s. gov't; research-article; journal article; \n",
      "isOpenAccess:\tY\n",
      "inEPMC:\tY\n",
      "inPMC:\tY\n",
      "hasPDF:\tY\n",
      "hasBook:\tN\n",
      "hasSuppl:\tY\n",
      "citedByCount:\t8\n",
      "hasReferences:\tY\n",
      "hasTextMinedTerms:\tY\n",
      "hasDbCrossReferences:\tY\n",
      "dbCrossReferenceList:\t{'dbName': ['EMBL']}\n",
      "hasLabsLinks:\tY\n",
      "hasTMAccessionNumbers:\tY\n",
      "tmAccessionTypeList:\t{'accessionType': ['gen']}\n",
      "firstPublicationDate:\t2012-12-12\n"
     ]
    }
   ],
   "source": [
    "for k, v in list(mgc_papers)[3].items():\n",
    "    print(k + ':\\t' + str(v))"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Slideshow",
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
